{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fbc9302a",
   "metadata": {},
   "source": [
    "# Visualisation & Application "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c86cba6",
   "metadata": {},
   "source": [
    "## Needed packages:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cc3d4b80",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import plotly.express as px\n",
    "import dash\n",
    "import dash_cytoscape as cyto\n",
    "import dash_bootstrap_components as dbc\n",
    "from dash import html\n",
    "from dash import dcc\n",
    "from dash.dependencies import Input, Output\n",
    "import json\n",
    "from time import perf_counter\n",
    "import coordinates\n",
    "import db_retrieve"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "023f53c2",
   "metadata": {},
   "source": [
    "## Some function:\n",
    "* to expand a dataframe\n",
    "* to load the dataframe into a graph"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1db108e4",
   "metadata": {},
   "source": [
    "### Expand the dataframe\n",
    "* wide dataframe\n",
    "* for every phenotype, the corresponding gene and ortholog"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e3ba17eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def expand_dataframe(df: pd.DataFrame):\n",
    "    \"\"\"Function to expand dataframe, so each phenotype is in a separate row.\n",
    "    Returns new dataframe and the total number of genes\n",
    "\n",
    "    :param df: dataframe to expand\n",
    "    \"\"\"\n",
    "    df_1 = df.assign(phenotype=df['Enriched_Phenotypes'].astype(str).str.split(',')).explode('Enriched_Phenotypes')\n",
    "    df_2 = df_1[['Orthlog_Genes', 'Human_Gene', 'Organism', 'Enriched_Phenotypes']]\n",
    "    # get total number of genes\n",
    "    n_genes = set(df['Human_Gene'].tolist())\n",
    "    n = len(n_genes)  # total number of genes\n",
    "\n",
    "    return df_2, n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dbc86774",
   "metadata": {},
   "source": [
    "### Load the dataframe into a graph (nodes and edges)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "63ca1263",
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_info_to_graph(dataframe: pd.DataFrame, metadata: pd.DataFrame, n_genes: int):\n",
    "    \"\"\" Function used to extract information from dataframe and load it into\n",
    "    nodes and edges list of dictionaries. For phenotype nodes, metadata is added as well\n",
    "\n",
    "    :param dataframe: dataframe with information about nodes and edges\n",
    "    :param metadata: dataframe with metadata of given pathway\n",
    "    :param n_genes: number of unique genes, necessary for y position determination\n",
    "    \"\"\"\n",
    "\n",
    "    # make nodes and edges\n",
    "    nodes_list = []\n",
    "    edges_list = []\n",
    "    nodes_set = set()  # to avoid duplication\n",
    "\n",
    "    # for node positions\n",
    "    count_n = 1  # count for x_pos of gene nodes\n",
    "    coordinates_ort = []\n",
    "    coordinates_phenotype = []\n",
    "\n",
    "    # Load data into nodes and edges\n",
    "    for index, row in dataframe.iterrows():\n",
    "        # source node is human gene\n",
    "        gene = str(row['Human_Gene'])\n",
    "        # target node is ortholog\n",
    "        ortholog = str(row['Orthlog_Genes'])\n",
    "        phenotype = str(row['Enriched_Phenotypes'])\n",
    "        organism = str(row['Organism'])\n",
    "        # add data and class info to the nodes\n",
    "        # gene node positions\n",
    "        # pos_x = 50  # always the same; set to 500 in case of large network\n",
    "        pos_x, pos_y = coordinates.gene_coordinate(1, n_genes, count_n)\n",
    "        cy_gene = {'data': {'id': gene,\n",
    "                            'indent': 'gene',   #add\n",
    "                            'label': gene,\n",
    "                            'size': 4,\n",
    "                            'fontsize': '1.5px'},\n",
    "                   'classes': 'blue',\n",
    "                   'selectable': True, #Allow selected\n",
    "                   'position': {'x': pos_x, 'y': pos_y}}\n",
    "\n",
    "        # ortholog node\n",
    "        pos_x_ortholog, pos_y_ortholog = coordinates.check_coordinates(coordinates_ort, pos_y, n_genes, 20, 45, 55, 80)\n",
    "        if ortholog != \"NaN\":\n",
    "            cy_ortholog = {'data': {'id': ortholog,\n",
    "                                    'label': ortholog,\n",
    "                                    'indent': 'ortholog', #add\n",
    "                                    'size': 2,\n",
    "                                    'fontsize': '1px',\n",
    "                                    'organism': organism},\n",
    "                           'classes': 'ortholog', #add\n",
    "                           'selectable': True,    #Allow selected\n",
    "                           'position': {'x': pos_x_ortholog, 'y': pos_y_ortholog}}\n",
    "            cy_edge = {\n",
    "                'data': {'id': gene + ortholog,\n",
    "                         'source': gene,\n",
    "                         'target': ortholog,\n",
    "                         'width': '0.25',\n",
    "                         'color': '#696969'}}\n",
    "\n",
    "        # add genes and orthologs to set nodes and cy nodes\n",
    "        if gene not in nodes_set:\n",
    "            nodes_set.add(gene)\n",
    "            nodes_list.append(cy_gene)\n",
    "            count_n += 1\n",
    "\n",
    "        if ortholog not in nodes_set:\n",
    "            nodes_set.add(ortholog)\n",
    "            nodes_list.append(cy_ortholog)\n",
    "        edges_list.append(cy_edge)\n",
    "\n",
    "        if phenotype != \"None\": # get rid of None node\n",
    "            metadata_phenotype = metadata.get(phenotype)  # load metadata\n",
    "            # phenotype node\n",
    "            pos_x_phenotype, pos_y_phenotype = coordinates.check_coordinates(coordinates_phenotype, pos_y, n_genes, 0, 20, 80, 100)\n",
    "\n",
    "            cy_phenotype = {'data': {'id': phenotype,\n",
    "                                     'label': phenotype,\n",
    "                                     'indent': 'phenotype',\n",
    "                                     'metadata': metadata_phenotype,\n",
    "                                     'size': 1,\n",
    "                                     'fontsize': '0.5px'},\n",
    "                            'classes': 'purple',\n",
    "                            'selectable': True,  #Allow selected\n",
    "                            'grabbable': False,   #Not allow grab for now\n",
    "                            'position': {'x': pos_x_phenotype, 'y': pos_y_phenotype}}\n",
    "            cy_edge_2 = {'data': {'id': ortholog + phenotype,\n",
    "                                  'source': ortholog,\n",
    "                                  'target': phenotype,\n",
    "                                  'width': '0.10',\n",
    "                                  'color': '#B8B8B8'}}\n",
    "\n",
    "            if phenotype not in nodes_set:\n",
    "                nodes_set.add(phenotype)\n",
    "                nodes_list.append(cy_phenotype)\n",
    "\n",
    "            edges_list.append(cy_edge_2)\n",
    "    return nodes_list, edges_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce41dccf",
   "metadata": {},
   "source": [
    "## Load Data:\n",
    "* The list of pathways we have so far:\n",
    "    * Phase2ConjugationOfCompounds (highest)\n",
    "    * Phase1CompundFunctionalization (1 level down)\n",
    "    * AmineOxidase (lowest)\n",
    "    * AminoAcidConjugation (lowest)\n",
    "    * EthanolOxidation (lowest)\n",
    "    * AHR (lowest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "28f3287a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Print each row and it's columns values\n",
      "PostgreSQL connection is closed\n",
      "PostgreSQL connection is closed\n"
     ]
    }
   ],
   "source": [
    "# upper level\n",
    "upper_df1 = db_retrieve.select_from_enrichment_results(\"Phase1CompundFunctionalization\")\n",
    "upper_df_1, upper_N_genes_1 = expand_dataframe(upper_df1)\n",
    "metadata = db_retrieve.select_from_metadata(\"Phase1CompundFunctionalization\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cd8d16c6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Print each row and it's columns values\n",
      "PostgreSQL connection is closed\n",
      "PostgreSQL connection is closed\n"
     ]
    }
   ],
   "source": [
    "#AHR\n",
    "gene_df1 = db_retrieve.select_from_enrichment_results(\"AHR\")               #gene_df1 (type = pandas.core.frame.DataFrame)\n",
    "gene_df_1, N_genes_1 = expand_dataframe(gene_df1) \n",
    "metadata1 = db_retrieve.select_from_metadata(\"AHR\")                       #metadata1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "419fc6ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Orthlog_Genes</th>\n",
       "      <th>Human_Gene</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Enriched_Phenotypes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arnt2</td>\n",
       "      <td>ENSG00000172379</td>\n",
       "      <td>zebrafish</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000288622</td>\n",
       "      <td>zebrafish</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>abnormal neuroanatomy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>abnormal neuroanatomy (with ss[D115.7])</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>abnormal neuroanatomy | somatic clone</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>antenna (with ss[aBR30-71])</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>chaeta (with ss[aBR30-71])</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>dendrite &amp; dorsal multidendritic neuron ddaB |...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FBgn0003513</td>\n",
       "      <td>ENSG00000063438</td>\n",
       "      <td>dmelanogaster</td>\n",
       "      <td>dendrite &amp; dorsal multidendritic neuron ddaD |...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Orthlog_Genes       Human_Gene       Organism  \\\n",
       "1         arnt2  ENSG00000172379      zebrafish   \n",
       "2           NaN  ENSG00000288622      zebrafish   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "3   FBgn0003513  ENSG00000063438  dmelanogaster   \n",
       "\n",
       "                                 Enriched_Phenotypes  \n",
       "1                                               None  \n",
       "2                                               None  \n",
       "3                              abnormal neuroanatomy  \n",
       "3            abnormal neuroanatomy (with ss[D115.7])  \n",
       "3              abnormal neuroanatomy | somatic clone  \n",
       "3                        antenna (with ss[aBR30-71])  \n",
       "3                         chaeta (with ss[aBR30-71])  \n",
       "3  dendrite & dorsal multidendritic neuron ddaB |...  \n",
       "3  dendrite & dorsal multidendritic neuron ddaD |...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# example of dataframe\n",
    "gene_df_1[1:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a24441eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Print each row and it's columns values\n",
      "PostgreSQL connection is closed\n",
      "PostgreSQL connection is closed\n"
     ]
    }
   ],
   "source": [
    "#AmineOxidase\n",
    "gene_df2 = db_retrieve.select_from_enrichment_results(\"AmineOxidase\")     #gene_df2\n",
    "gene_df_2, N_genes_2 = expand_dataframe(gene_df2) \n",
    "metadata2 = db_retrieve.select_from_metadata(\"AmineOxidase\")              #metadata2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d2b382c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Print each row and it's columns values\n",
      "PostgreSQL connection is closed\n",
      "PostgreSQL connection is closed\n"
     ]
    }
   ],
   "source": [
    "#AminoAcidConjugation\n",
    "gene_df3 = db_retrieve.select_from_enrichment_results(\"AminoAcidConjugation\")     #gene_df3\n",
    "gene_df_3, N_genes_3 = expand_dataframe(gene_df3) \n",
    "metadata3 = db_retrieve.select_from_metadata(\"AminoAcidConjugation\")              #metadata3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "72490ca4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Print each row and it's columns values\n",
      "PostgreSQL connection is closed\n",
      "PostgreSQL connection is closed\n"
     ]
    }
   ],
   "source": [
    "#EthanolOxidation\n",
    "gene_df4 = db_retrieve.select_from_enrichment_results(\"EthanolOxidation\")     #gene_df4\n",
    "gene_df_4, N_genes_4 = expand_dataframe(gene_df4) \n",
    "metadata4 = db_retrieve.select_from_metadata(\"EthanolOxidation\")              #metadata4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1feea068",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_list = ['AHR', 'Amino Oxidase', 'Ethanol Oxidation', 'AminoAcid Conjugation']  #add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a5b913fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#nodes & edges\n",
    "nodes, edges = load_info_to_graph(upper_df_1, metadata, upper_N_genes_1)\n",
    "nodes1, edges1 = load_info_to_graph(gene_df_1, metadata1, N_genes_1)\n",
    "nodes2, edges2 = load_info_to_graph(gene_df_2, metadata2, N_genes_2)\n",
    "nodes3, edges3 = load_info_to_graph(gene_df_3, metadata3, N_genes_3)\n",
    "nodes4, edges4 = load_info_to_graph(gene_df_4, metadata4, N_genes_4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0864f15e",
   "metadata": {},
   "outputs": [],
   "source": [
    "element = edges + nodes\n",
    "new_element1 = edges1 + nodes1\n",
    "new_element2 = edges2 + nodes2\n",
    "new_element3 = edges3 + nodes3\n",
    "new_element4 = edges4 + nodes4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0a368f65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'data': {'id': 'ENSG00000143437',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000143437',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 12.5}},\n",
       " {'data': {'id': 'arnt',\n",
       "   'label': 'arnt',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 41.83495621677152, 'y': 48.064603068407585}},\n",
       " {'data': {'id': 'ENSG00000172379',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000172379',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 25.0}},\n",
       " {'data': {'id': 'arnt2',\n",
       "   'label': 'arnt2',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 39.55501146034899, 'y': 64.72956443328935}},\n",
       " {'data': {'id': 'ENSG00000288622',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000288622',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 37.5}},\n",
       " {'data': {'id': 'arnt2',\n",
       "   'label': 'arnt2',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 39.55501146034899, 'y': 64.72956443328935}},\n",
       " {'data': {'id': 'ENSG00000063438',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000063438',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 50.0}},\n",
       " {'data': {'id': 'FBgn0003513',\n",
       "   'label': 'FBgn0003513',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 77.13848822869723, 'y': 80.33769421718459}},\n",
       " {'data': {'id': 'abnormal neuroanatomy',\n",
       "   'label': 'abnormal neuroanatomy',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 344.0, 0.076],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.627481996665919, 'y': 67.09365190368078}},\n",
       " {'data': {'id': 'abnormal neuroanatomy (with ss[D115.7])',\n",
       "   'label': 'abnormal neuroanatomy (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 86.0, 0.019],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.77826792041729, 'y': 9.85069063661741}},\n",
       " {'data': {'id': 'abnormal neuroanatomy | somatic clone',\n",
       "   'label': 'abnormal neuroanatomy | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.042, 414.0, 0.091],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.285689694262276, 'y': 65.63242388578729}},\n",
       " {'data': {'id': 'antenna (with ss[aBR30-71])',\n",
       "   'label': 'antenna (with ss[aBR30-71])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 123.0, 0.027],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.432706589325917, 'y': 21.84952843342185}},\n",
       " {'data': {'id': 'chaeta (with ss[aBR30-71])',\n",
       "   'label': 'chaeta (with ss[aBR30-71])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 125.0, 0.027],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.332923979061945, 'y': 56.74315162587461}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaB | somatic clone',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaB | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 132.0, 0.029],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.44630120486359, 'y': 67.91909903178679}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaD | somatic clone',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaD | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 126.0, 0.028],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.138369874491303, 'y': 58.09278003023892}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaE',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaE',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 204.0, 0.045],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.57611949040046, 'y': 58.142064335803646}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaE (with ss[D115.7])',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaE (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 127.0, 0.028],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.78937891183467, 'y': 52.88090515725861}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaE | somatic clone',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaE | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 205.0, 0.045],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.4586086826414335, 'y': 67.84925850350618}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaF | somatic clone',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaF | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 128.0, 0.028],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.56624851184852, 'y': 51.763680143086255}},\n",
       " {'data': {'id': 'dendrite & lateral multidendritic neuron ldaB | somatic clone',\n",
       "   'label': 'dendrite & lateral multidendritic neuron ldaB | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 129.0, 0.028],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.831631585355531, 'y': 29.648407159095903}},\n",
       " {'data': {'id': 'leg | ectopic (with ss[aBR30-71])',\n",
       "   'label': 'leg | ectopic (with ss[aBR30-71])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 79.0, 0.017],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.39072971365098, 'y': 59.089019593899295}},\n",
       " {'data': {'id': 'lethal | recessive',\n",
       "   'label': 'lethal | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.065, 428.0, 0.094],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.43915285017347, 'y': 96.02026703700591}},\n",
       " {'data': {'id': 'visible (with ss[aBR30-71])',\n",
       "   'label': 'visible (with ss[aBR30-71])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 131.0, 0.029],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.30146924574895, 'y': 60.93737344260241}},\n",
       " {'data': {'id': 'chaeta (with ss[aP88])',\n",
       "   'label': 'chaeta (with ss[aP88])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 80.0, 0.018],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.09534599482305, 'y': 32.89987018314397}},\n",
       " {'data': {'id': 'viable',\n",
       "   'label': 'viable',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 190.0, 0.042],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.57931322980318, 'y': 87.27501419556233}},\n",
       " {'data': {'id': 'visible (with ss[aP88])',\n",
       "   'label': 'visible (with ss[aP88])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 115.0, 0.025],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.6819477879532263, 'y': 87.03246305987827}},\n",
       " {'data': {'id': 'antenna',\n",
       "   'label': 'antenna',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.026, 396.0, 0.087],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.68017700104058, 'y': 25.24750595825297}},\n",
       " {'data': {'id': 'visible | homeotic',\n",
       "   'label': 'visible | homeotic',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 347.0, 0.076],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.60938402466529, 'y': 44.89036681377371}},\n",
       " {'data': {'id': 'posterior vertical bristle',\n",
       "   'label': 'posterior vertical bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 311.0, 0.068],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.6042405194587457, 'y': 14.854324229354887}},\n",
       " {'data': {'id': 'scutellar bristle',\n",
       "   'label': 'scutellar bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.038, 413.0, 0.091],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.922567219060263, 'y': 43.965167215227986}},\n",
       " {'data': {'id': 'vertical bristle',\n",
       "   'label': 'vertical bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 233.0, 0.051],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.657088923635618, 'y': 1.0106599145352502}},\n",
       " {'data': {'id': 'tarsal segment',\n",
       "   'label': 'tarsal segment',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.023, 389.0, 0.085],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.988315903672, 'y': 56.55752960232259}},\n",
       " {'data': {'id': 'arista (with Df(3R)ss-D114.4)',\n",
       "   'label': 'arista (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 78.0, 0.017],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.2170179038605, 'y': 56.809627331897914}},\n",
       " {'data': {'id': 'tarsal segment | ectopic (with Df(3R)ss-D114.4)',\n",
       "   'label': 'tarsal segment | ectopic (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 63.0, 0.014],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.11452507350371, 'y': 71.51383016417404}},\n",
       " {'data': {'id': 'third segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'label': 'third segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 122.0, 0.027],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 14.399150185208187, 'y': 4.101935396305123}},\n",
       " {'data': {'id': 'visible (with Df(3R)ss-D114.4)',\n",
       "   'label': 'visible (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 26.0, 0.006],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.33384340223222, 'y': 31.849889713384027}},\n",
       " {'data': {'id': 'antennal disc (with Df(3R)ss-D114.4)',\n",
       "   'label': 'antennal disc (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 120.0, 0.026],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.07585527454769, 'y': 50.022515836671985}},\n",
       " {'data': {'id': 'fifth segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'label': 'fifth segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 141.0, 0.031],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.72273770413366, 'y': 39.38806326106234}},\n",
       " {'data': {'id': 'fourth segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'label': 'fourth segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 142.0, 0.031],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.609178177415352, 'y': 4.359249054043291}},\n",
       " {'data': {'id': 'leg joint (with ss[a])',\n",
       "   'label': 'leg joint (with ss[a])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 143.0, 0.031],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.028272899180404, 'y': 9.986138869788608}},\n",
       " {'data': {'id': 'leg | ectopic',\n",
       "   'label': 'leg | ectopic',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 296.0, 0.065],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.940809493372601, 'y': 61.96723218299809}},\n",
       " {'data': {'id': 'metatarsus (with ss[a])',\n",
       "   'label': 'metatarsus (with ss[a])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 144.0, 0.032],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.33109182394952974, 'y': 41.57872304325475}},\n",
       " {'data': {'id': 'pretarsus (with ss[D115.7])',\n",
       "   'label': 'pretarsus (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 145.0, 0.032],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.935872635561109, 'y': 5.418785394082737}},\n",
       " {'data': {'id': 'second segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'label': 'second segment of antenna (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 146.0, 0.032],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.1566353345981661, 'y': 10.046727262423383}},\n",
       " {'data': {'id': 'sex comb | ectopic (with ss[a])',\n",
       "   'label': 'sex comb | ectopic (with ss[a])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 147.0, 0.032],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.111886610120596, 'y': 17.01595920085768}},\n",
       " {'data': {'id': 'tarsal segment 2 (with ss[D115.7])',\n",
       "   'label': 'tarsal segment 2 (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 148.0, 0.033],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.83335928450856, 'y': 75.2120416972143}},\n",
       " {'data': {'id': 'tarsal segment 3 (with ss[D115.7])',\n",
       "   'label': 'tarsal segment 3 (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 149.0, 0.033],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.278313728120594, 'y': 51.57656606042204}},\n",
       " {'data': {'id': 'tarsal segment 4 (with ss[D115.7])',\n",
       "   'label': 'tarsal segment 4 (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 150.0, 0.033],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.580638583438077, 'y': 35.14270427557865}},\n",
       " {'data': {'id': 'ventral thoracic disc (with ss[D115.7])',\n",
       "   'label': 'ventral thoracic disc (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 135.0, 0.03],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.9380837737393053, 'y': 56.14341707279878}},\n",
       " {'data': {'id': 'visible',\n",
       "   'label': 'visible',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.031, 406.0, 0.089],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.425229143794308, 'y': 1.4941379388216491}},\n",
       " {'data': {'id': 'visible (with ss[D115.7])',\n",
       "   'label': 'visible (with ss[D115.7])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 134.0, 0.029],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.974016746194913, 'y': 40.500720406869625}},\n",
       " {'data': {'id': 'abnormal neuroanatomy (with ss[134])',\n",
       "   'label': 'abnormal neuroanatomy (with ss[134])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 133.0, 0.029],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.692769412883, 'y': 88.69476769874836}},\n",
       " {'data': {'id': 'antennal sense organ',\n",
       "   'label': 'antennal sense organ',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 321.0, 0.071],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.361466719633443, 'y': 98.59703812039962}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaC | somatic clone',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaC | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 278.0, 0.061],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.8033946857042, 'y': 61.759239746925196}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaE (with ss[134])',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaE (with ss[134])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 124.0, 0.027],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.59226713208723, 'y': 78.6655861275978}},\n",
       " {'data': {'id': 'dorsomedial papilla',\n",
       "   'label': 'dorsomedial papilla',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 217.0, 0.048],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.55185446311131, 'y': 10.519907068142375}},\n",
       " {'data': {'id': 'leg',\n",
       "   'label': 'leg',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.065, 425.0, 0.093],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.371333882883387, 'y': 2.1721533580797314}},\n",
       " {'data': {'id': 'maxillary palpus',\n",
       "   'label': 'maxillary palpus',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 343.0, 0.075],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.15603033154282, 'y': 21.413824955848217}},\n",
       " {'data': {'id': 'metatarsus',\n",
       "   'label': 'metatarsus',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 304.0, 0.067],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.958227789258464, 'y': 59.84479373458097}},\n",
       " {'data': {'id': 'pale ommatidium | ectopic',\n",
       "   'label': 'pale ommatidium | ectopic',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 118.0, 0.026],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.44542685716888, 'y': 58.66927715326034}},\n",
       " {'data': {'id': 'pale ommatidium | ectopic | somatic clone',\n",
       "   'label': 'pale ommatidium | ectopic | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 119.0, 0.026],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.528295564872575, 'y': 60.05750777304469}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of pale ommatidium | ectopic | somatic clone',\n",
       "   'label': 'photoreceptor cell R7 of pale ommatidium | ectopic | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 121.0, 0.027],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.8162391596208, 'y': 66.1489836796995}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of yellow ommatidium | somatic clone',\n",
       "   'label': 'photoreceptor cell R7 of yellow ommatidium | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 27.0, 0.006],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.158988886724881, 'y': 57.96641246102885}},\n",
       " {'data': {'id': 'pretarsus (with ss[D114.9])',\n",
       "   'label': 'pretarsus (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 28.0, 0.006],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.90068611172642, 'y': 97.94460353035028}},\n",
       " {'data': {'id': 'pretarsus | ectopic (with Df(3R)ss-D114.4)',\n",
       "   'label': 'pretarsus | ectopic (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 29.0, 0.006],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.041092403034462, 'y': 52.722122104076185}},\n",
       " {'data': {'id': 'tarsal segment (with Df(3R)ss-D114.4)',\n",
       "   'label': 'tarsal segment (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 11.0, 0.002],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.98647531326242, 'y': 15.911339576758088}},\n",
       " {'data': {'id': 'tarsal segment 2',\n",
       "   'label': 'tarsal segment 2',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 307.0, 0.067],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.392150706079722, 'y': 4.755173390602064}},\n",
       " {'data': {'id': 'tarsal segment 2 (with ss[D114.9])',\n",
       "   'label': 'tarsal segment 2 (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 12.0, 0.003],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.82561305016137, 'y': 95.42400385472705}},\n",
       " {'data': {'id': 'tarsal segment 3',\n",
       "   'label': 'tarsal segment 3',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 325.0, 0.071],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.53052816180578, 'y': 65.38351962001425}},\n",
       " {'data': {'id': 'tarsal segment 3 (with ss[D114.9])',\n",
       "   'label': 'tarsal segment 3 (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 22.0, 0.005],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.30728325998133, 'y': 35.91135533509462}},\n",
       " {'data': {'id': 'tarsal segment 4',\n",
       "   'label': 'tarsal segment 4',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 336.0, 0.074],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.82454097737893, 'y': 16.376777944574407}},\n",
       " {'data': {'id': 'tarsal segment 4 (with ss[D114.9])',\n",
       "   'label': 'tarsal segment 4 (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 15.0, 0.003],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.644806779158716, 'y': 54.29165392795065}},\n",
       " {'data': {'id': 'unguis | ectopic (with Df(3R)ss-D114.4)',\n",
       "   'label': 'unguis | ectopic (with Df(3R)ss-D114.4)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 16.0, 0.004],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.271403222116263, 'y': 36.27557033880565}},\n",
       " {'data': {'id': 'ventral thoracic disc (with ss[D114.9])',\n",
       "   'label': 'ventral thoracic disc (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 17.0, 0.004],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.404519895409821, 'y': 64.71933632893744}},\n",
       " {'data': {'id': 'visible (with ss[D114.9])',\n",
       "   'label': 'visible (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 18.0, 0.004],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.8496685503824, 'y': 16.33841437005341}},\n",
       " {'data': {'id': 'yellow ommatidium',\n",
       "   'label': 'yellow ommatidium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 202.0, 0.044],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.699094006978454, 'y': 24.927676355339646}},\n",
       " {'data': {'id': 'yellow ommatidium | somatic clone',\n",
       "   'label': 'yellow ommatidium | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 19.0, 0.004],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.852698612154253, 'y': 4.78249499483705}},\n",
       " {'data': {'id': 'trichogen cell, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'label': 'trichogen cell, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.058, 422.0, 0.093],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.93027522263345, 'y': 12.575509791609896}},\n",
       " {'data': {'id': 'viable, with Dcr-2[UAS.cDa], Scer\\\\GAL4[elav.PLu]',\n",
       "   'label': 'viable, with Dcr-2[UAS.cDa], Scer\\\\GAL4[elav.PLu]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 297.0, 0.065],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.30199945323468, 'y': 94.11584503519501}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 248.0, 0.055],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.148251366377856, 'y': 62.19897402118973}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.069, 430.0, 0.095],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.35674090154414, 'y': 46.98655384396704}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[tin.CΔ4]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[tin.CΔ4]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.02, 383.0, 0.084],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.883907572952609, 'y': 90.51946733514866}},\n",
       " {'data': {'id': 'abnormal gravitaxis',\n",
       "   'label': 'abnormal gravitaxis',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.009, 354.0, 0.078],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.477971231848693, 'y': 2.671120059138987}},\n",
       " {'data': {'id': 'arista',\n",
       "   'label': 'arista',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.027, 398.0, 0.087],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.045561655772483, 'y': 78.6134118152697}},\n",
       " {'data': {'id': 'abnormal neuroanatomy, with Scer\\\\GAL4[477]',\n",
       "   'label': 'abnormal neuroanatomy, with Scer\\\\GAL4[477]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 280.0, 0.062],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.20247579011576, 'y': 58.005142549065006}},\n",
       " {'data': {'id': 'abnormal neuroanatomy, with Scer\\\\GAL4[ppk.PG]',\n",
       "   'label': 'abnormal neuroanatomy, with Scer\\\\GAL4[ppk.PG]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.01, 357.0, 0.078],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.1715214421089204, 'y': 69.32762156500992}},\n",
       " {'data': {'id': 'dendrite & dorsal multidendritic neuron ddaC, with Scer\\\\GAL4[ppk.PG]',\n",
       "   'label': 'dendrite & dorsal multidendritic neuron ddaC, with Scer\\\\GAL4[ppk.PG]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 20.0, 0.004],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.20292834027573425, 'y': 54.046467865262194}},\n",
       " {'data': {'id': 'dendrite & tracheal innervating neuron, with Scer\\\\GAL4[477]',\n",
       "   'label': 'dendrite & tracheal innervating neuron, with Scer\\\\GAL4[477]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 21.0, 0.005],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.56941680578856, 'y': 34.27140726160346}},\n",
       " {'data': {'id': 'denticle belt | medial, with Scer\\\\GAL4[69B]',\n",
       "   'label': 'denticle belt | medial, with Scer\\\\GAL4[69B]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 39.0, 0.009],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.30006891339164, 'y': 14.36308264423346}},\n",
       " {'data': {'id': 'eye photoreceptor cell | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'label': 'eye photoreceptor cell | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 41.0, 0.009],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.337079955119008, 'y': 40.71762806597714}},\n",
       " {'data': {'id': 'eye photoreceptor cell, with Scer\\\\GAL4[GMR.PF]',\n",
       "   'label': 'eye photoreceptor cell, with Scer\\\\GAL4[GMR.PF]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.014, 369.0, 0.081],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.958896743448715, 'y': 4.920499835389669}},\n",
       " {'data': {'id': 'eye photoreceptor cell, with Scer\\\\GAL4[GMR.long]',\n",
       "   'label': 'eye photoreceptor cell, with Scer\\\\GAL4[GMR.long]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 258.0, 0.057],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.29026650389428, 'y': 45.801149953050626}},\n",
       " {'data': {'id': 'eye, with Scer\\\\GAL4[GMR.long]',\n",
       "   'label': 'eye, with Scer\\\\GAL4[GMR.long]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 334.0, 0.073],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.0032538724109398842, 'y': 84.39797935284214}},\n",
       " {'data': {'id': 'femur | distal',\n",
       "   'label': 'femur | distal',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 10.0, 0.002],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.4130961528358252, 'y': 43.81441548934396}},\n",
       " {'data': {'id': 'increased cell death | cell autonomous | somatic clone, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'label': 'increased cell death | cell autonomous | somatic clone, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 9.0, 0.002],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.86349359774572, 'y': 69.53856404188483}},\n",
       " {'data': {'id': 'increased cell death | cell non-autonomous | somatic clone, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'label': 'increased cell death | cell non-autonomous | somatic clone, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 8.0, 0.002],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.89102602632269, 'y': 93.32160756243798}},\n",
       " {'data': {'id': 'leg anterior compartment',\n",
       "   'label': 'leg anterior compartment',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 250.0, 0.055],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.026667369190339, 'y': 60.61045385998774}},\n",
       " {'data': {'id': 'leg | distal, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'label': 'leg | distal, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 30.0, 0.007],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.231948723179737, 'y': 68.94718514590394}},\n",
       " {'data': {'id': 'lethal | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'label': 'lethal | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 293.0, 0.064],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.796833764729865, 'y': 31.787533698745875}},\n",
       " {'data': {'id': 'maxillary palpus, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'label': 'maxillary palpus, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 31.0, 0.007],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.649427523069388, 'y': 88.7694978598824}},\n",
       " {'data': {'id': 'ommatidium, with Scer\\\\GAL4[GMR.long]',\n",
       "   'label': 'ommatidium, with Scer\\\\GAL4[GMR.long]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 292.0, 0.064],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.8449432344028924, 'y': 78.06821314000413}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of pale ommatidium | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'label': 'photoreceptor cell R7 of pale ommatidium | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 33.0, 0.007],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.2798159184962, 'y': 69.0478971334994}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of pale ommatidium, with Scer\\\\GAL4[GMR.PF]',\n",
       "   'label': 'photoreceptor cell R7 of pale ommatidium, with Scer\\\\GAL4[GMR.PF]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 40.0, 0.009],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.51363208249644, 'y': 35.024659936767755}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of pale ommatidium, with Scer\\\\GAL4[Pan-R7]',\n",
       "   'label': 'photoreceptor cell R7 of pale ommatidium, with Scer\\\\GAL4[Pan-R7]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 34.0, 0.007],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.166516788869963, 'y': 68.93032015433695}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of yellow ommatidium | ectopic | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'label': 'photoreceptor cell R7 of yellow ommatidium | ectopic | heat sensitive, with Scer\\\\GAL4[hs.PB]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 35.0, 0.008],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.59405109682132, 'y': 24.19654078855823}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of yellow ommatidium | ectopic, with Scer\\\\GAL4[GMR.PF]',\n",
       "   'label': 'photoreceptor cell R7 of yellow ommatidium | ectopic, with Scer\\\\GAL4[GMR.PF]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 36.0, 0.008],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.976579217206524, 'y': 88.77136652198259}},\n",
       " {'data': {'id': 'photoreceptor cell R7 of yellow ommatidium | ectopic, with Scer\\\\GAL4[Pan-R7]',\n",
       "   'label': 'photoreceptor cell R7 of yellow ommatidium | ectopic, with Scer\\\\GAL4[Pan-R7]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 37.0, 0.008],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.399088702714966, 'y': 46.342380496834664}},\n",
       " {'data': {'id': 'rostral membrane, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'label': 'rostral membrane, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 38.0, 0.008],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.00084857562867, 'y': 59.760121706391665}},\n",
       " {'data': {'id': 'tergite, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'label': 'tergite, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 24.0, 0.005],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.0431466062536785, 'y': 29.125925936380103}},\n",
       " {'data': {'id': 'tibia, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'label': 'tibia, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 25.0, 0.005],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.36888233010802, 'y': 4.323301145217073}},\n",
       " {'data': {'id': 'tracheal branch primordium, with Scer\\\\GAL4[69B]',\n",
       "   'label': 'tracheal branch primordium, with Scer\\\\GAL4[69B]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 195.0, 0.043],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.94954268273033, 'y': 53.17990891850791}},\n",
       " {'data': {'id': 'wing disc | cell autonomous | somatic clone, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'label': 'wing disc | cell autonomous | somatic clone, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 32.0, 0.007],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.17313058478786, 'y': 81.72394044474862}},\n",
       " {'data': {'id': 'wing margin, with Scer\\\\GAL4[NP7148]',\n",
       "   'label': 'wing margin, with Scer\\\\GAL4[NP7148]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 200.0, 0.044],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.02111715177128, 'y': 68.9815935634557}},\n",
       " {'data': {'id': 'wing, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'label': 'wing, with Scer\\\\GAL4[ptc-559.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.02, 382.0, 0.084],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.7477762253684, 'y': 37.71423940024351}},\n",
       " {'data': {'id': 'wing, with Scer\\\\GAL4[sd-SG29.1]',\n",
       "   'label': 'wing, with Scer\\\\GAL4[sd-SG29.1]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.027, 400.0, 0.088],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.41021151194953, 'y': 19.50422724022305}},\n",
       " {'data': {'id': 'eye, with Scer\\\\GAL4[GMR.PU]',\n",
       "   'label': 'eye, with Scer\\\\GAL4[GMR.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 273.0, 0.06],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.89220986913814, 'y': 52.449376119456225}},\n",
       " {'data': {'id': 'leg, with Scer\\\\GAL4[rn-GAL4]',\n",
       "   'label': 'leg, with Scer\\\\GAL4[rn-GAL4]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 23.0, 0.005],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.6079238330265, 'y': 79.91282404296803}},\n",
       " {'data': {'id': 'tarsal segment, with Scer\\\\GAL4[rn-GAL4]',\n",
       "   'label': 'tarsal segment, with Scer\\\\GAL4[rn-GAL4]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 14.0, 0.003],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.556789258568426, 'y': 96.25142591168732}},\n",
       " {'data': {'id': 'visible, with Scer\\\\GAL4[GMR.PU]',\n",
       "   'label': 'visible, with Scer\\\\GAL4[GMR.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.042, 415.0, 0.091],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.820300452937, 'y': 66.53558604477627}},\n",
       " {'data': {'id': 'visible, with Scer\\\\GAL4[rn-GAL4]',\n",
       "   'label': 'visible, with Scer\\\\GAL4[rn-GAL4]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 7.0, 0.002],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.80655681914376, 'y': 34.22290706691995}},\n",
       " {'data': {'id': 'abnormal courtship behavior',\n",
       "   'label': 'abnormal courtship behavior',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.016, 376.0, 0.083],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.39828528266985, 'y': 30.007492637274414}},\n",
       " {'data': {'id': 'abnormal memory',\n",
       "   'label': 'abnormal memory',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.035, 410.0, 0.09],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.27991003173949, 'y': 37.50381416028551}},\n",
       " {'data': {'id': 'flightless | recessive',\n",
       "   'label': 'flightless | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.014, 370.0, 0.081],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.77406949114072, 'y': 30.556578665317026}},\n",
       " {'data': {'id': 'mesothoracic tarsal segment | distal | ectopic',\n",
       "   'label': 'mesothoracic tarsal segment | distal | ectopic',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 140.0, 0.031],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.34830693200121, 'y': 83.24651576999284}},\n",
       " {'data': {'id': 'visible | homeotic | recessive',\n",
       "   'label': 'visible | homeotic | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 265.0, 0.058],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.13301223531772, 'y': 75.9467113587958}},\n",
       " {'data': {'id': 'antennal segment',\n",
       "   'label': 'antennal segment',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 236.0, 0.052],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.73601466727222, 'y': 4.826082962538836}},\n",
       " {'data': {'id': 'mesothoracic tarsal segment',\n",
       "   'label': 'mesothoracic tarsal segment',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 223.0, 0.049],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.68418658762235, 'y': 42.23814853902671}},\n",
       " {'data': {'id': 'metathoracic tarsal segment',\n",
       "   'label': 'metathoracic tarsal segment',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 279.0, 0.061],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.38083320978778, 'y': 28.67996900133223}},\n",
       " {'data': {'id': 'prothoracic tarsal segment',\n",
       "   'label': 'prothoracic tarsal segment',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 230.0, 0.051],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.8912082573099624, 'y': 50.50872348766995}},\n",
       " {'data': {'id': 'female semi-sterile',\n",
       "   'label': 'female semi-sterile',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 1.0, 0.0],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.94595359851478, 'y': 51.687741374818394}},\n",
       " {'data': {'id': 'male sterile',\n",
       "   'label': 'male sterile',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.01, 355.0, 0.078],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.810947074974154, 'y': 59.38467268707788}},\n",
       " {'data': {'id': 'microchaeta',\n",
       "   'label': 'microchaeta',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.043, 416.0, 0.091],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.927138966604431, 'y': 78.53307147301744}},\n",
       " {'data': {'id': 'sex comb',\n",
       "   'label': 'sex comb',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.017, 379.0, 0.083],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.264929742079186, 'y': 95.03882053202511}},\n",
       " {'data': {'id': 'unguis | ectopic',\n",
       "   'label': 'unguis | ectopic',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 211.0, 0.046],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.52401435962116, 'y': 20.53640471985505}},\n",
       " {'data': {'id': 'leg joint',\n",
       "   'label': 'leg joint',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 319.0, 0.07],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.36584616934871, 'y': 1.4999620028390281}},\n",
       " {'data': {'id': 'partially lethal',\n",
       "   'label': 'partially lethal',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.078, 434.0, 0.095],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.85606466800846, 'y': 72.92866317777785}},\n",
       " {'data': {'id': 'sterile',\n",
       "   'label': 'sterile',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.03, 405.0, 0.089],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.25250859497383, 'y': 55.087404703845976}},\n",
       " {'data': {'id': 'antenna (with ss[134])',\n",
       "   'label': 'antenna (with ss[134])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 109.0, 0.024],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.129216768625003, 'y': 47.368215593726624}},\n",
       " {'data': {'id': 'chaeta (with ss[134])',\n",
       "   'label': 'chaeta (with ss[134])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 110.0, 0.024],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.532967073389122, 'y': 64.65374807299948}},\n",
       " {'data': {'id': 'leg | ectopic (with ss[134])',\n",
       "   'label': 'leg | ectopic (with ss[134])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 111.0, 0.024],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.15574707893352, 'y': 66.08622431350443}},\n",
       " {'data': {'id': 'visible (with ss[134])',\n",
       "   'label': 'visible (with ss[134])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 112.0, 0.025],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.882714684191763, 'y': 63.4391961195702}},\n",
       " {'data': {'id': 'arista | proximal',\n",
       "   'label': 'arista | proximal',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 213.0, 0.047],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.403204742214022, 'y': 55.20959630787262}},\n",
       " {'data': {'id': 'third segment of antenna | distal',\n",
       "   'label': 'third segment of antenna | distal',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 214.0, 0.047],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.96163427113628, 'y': 87.98069282707006}},\n",
       " {'data': {'id': 'arista | cold sensitive',\n",
       "   'label': 'arista | cold sensitive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 215.0, 0.047],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.22306511205173, 'y': 95.05924810652078}},\n",
       " {'data': {'id': 'fertile | recessive',\n",
       "   'label': 'fertile | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 274.0, 0.06],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.03696031636997, 'y': 33.64389834660354}},\n",
       " {'data': {'id': 'viable | recessive',\n",
       "   'label': 'viable | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.009, 353.0, 0.078],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.8529279257005, 'y': 92.37629379068605}},\n",
       " {'data': {'id': 'chaeta (with ss[1])',\n",
       "   'label': 'chaeta (with ss[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 113.0, 0.025],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.1198158377247, 'y': 0.5169991147984554}},\n",
       " {'data': {'id': 'third segment of antenna',\n",
       "   'label': 'third segment of antenna',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 324.0, 0.071],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.646479994402547, 'y': 53.03798516472187}},\n",
       " {'data': {'id': 'visible (with ss[1])',\n",
       "   'label': 'visible (with ss[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 108.0, 0.024],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.74829241143019, 'y': 26.892779741675987}},\n",
       " {'data': {'id': 'abnormal size',\n",
       "   'label': 'abnormal size',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.014, 367.0, 0.081],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.731749533148191, 'y': 87.2435035540437}},\n",
       " {'data': {'id': 'decreased occurrence of cell division',\n",
       "   'label': 'decreased occurrence of cell division',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 322.0, 0.071],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.36886840905716, 'y': 20.34887321926743}},\n",
       " {'data': {'id': 'increased occurrence of cell division',\n",
       "   'label': 'increased occurrence of cell division',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 283.0, 0.062],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.8625938320271742, 'y': 84.11899212807538}},\n",
       " {'data': {'id': 'mesothoracic leg disc | larval stage',\n",
       "   'label': 'mesothoracic leg disc | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 245.0, 0.054],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.3966928668532, 'y': 42.56040978526905}},\n",
       " {'data': {'id': 'metathoracic leg disc | larval stage',\n",
       "   'label': 'metathoracic leg disc | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 220.0, 0.048],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.8486702581967, 'y': 65.8393194001573}},\n",
       " {'data': {'id': 'prothoracic leg disc | larval stage',\n",
       "   'label': 'prothoracic leg disc | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 244.0, 0.054],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.75460769544652, 'y': 28.930672535882717}},\n",
       " {'data': {'id': 'fertile',\n",
       "   'label': 'fertile',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 342.0, 0.075],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.99426298510257, 'y': 48.673580774687174}},\n",
       " {'data': {'id': 'thorax',\n",
       "   'label': 'thorax',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.023, 388.0, 0.085],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.0447877158987, 'y': 25.980477031111015}},\n",
       " {'data': {'id': 'abnormal behavior',\n",
       "   'label': 'abnormal behavior',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 3.0, 0.001],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.118309346849781, 'y': 14.419182552649868}},\n",
       " {'data': {'id': 'antenna | distal | somatic clone',\n",
       "   'label': 'antenna | distal | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 216.0, 0.047],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.2071520324101, 'y': 68.70274085881313}},\n",
       " {'data': {'id': 'eye-antennal disc',\n",
       "   'label': 'eye-antennal disc',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.016, 375.0, 0.082],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.140801564824082, 'y': 90.39555087152695}},\n",
       " {'data': {'id': 'leg joint (with ss[D114.9])',\n",
       "   'label': 'leg joint (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 99.0, 0.022],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.05287055354055, 'y': 15.515841995106172}},\n",
       " {'data': {'id': 'leg | ectopic | somatic clone',\n",
       "   'label': 'leg | ectopic | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 285.0, 0.063],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.994767307522725, 'y': 60.71782976620334}},\n",
       " {'data': {'id': 'metatarsus (with ss[D114.9])',\n",
       "   'label': 'metatarsus (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 100.0, 0.022],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.9290447258375689, 'y': 65.57939311191335}},\n",
       " {'data': {'id': 'sex comb | ectopic (with ss[D114.9])',\n",
       "   'label': 'sex comb | ectopic (with ss[D114.9])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 101.0, 0.022],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.48969576512214, 'y': 40.35823356187611}},\n",
       " {'data': {'id': 'tarsal segment | ectopic',\n",
       "   'label': 'tarsal segment | ectopic',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 261.0, 0.057],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.27923060351951, 'y': 2.2580702477422387}},\n",
       " {'data': {'id': 'tarsal segment (with ss[sta])',\n",
       "   'label': 'tarsal segment (with ss[sta])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 102.0, 0.022],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.36432118233873, 'y': 16.575377505000834}},\n",
       " {'data': {'id': 'visible (with ss[sta])',\n",
       "   'label': 'visible (with ss[sta])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 103.0, 0.023],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.78076417645426, 'y': 19.769418295095885}},\n",
       " {'data': {'id': 'mesothoracic leg',\n",
       "   'label': 'mesothoracic leg',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 338.0, 0.074],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.184937985013935, 'y': 50.62183086341337}},\n",
       " {'data': {'id': 'abnormal visual perception',\n",
       "   'label': 'abnormal visual perception',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 238.0, 0.052],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.86236479320142, 'y': 74.45575826988645}},\n",
       " {'data': {'id': 'tarsal segment (with ss[abr])',\n",
       "   'label': 'tarsal segment (with ss[abr])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 104.0, 0.023],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.3084313584109126, 'y': 65.10324536206211}},\n",
       " {'data': {'id': 'visible (with ss[abr])',\n",
       "   'label': 'visible (with ss[abr])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 105.0, 0.023],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.63072192366555, 'y': 54.05815786200356}},\n",
       " {'data': {'id': 'abnormal cell number | adult stage',\n",
       "   'label': 'abnormal cell number | adult stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 282.0, 0.062],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.10381048616645, 'y': 19.48700401976816}},\n",
       " {'data': {'id': 'abnormal neuroanatomy | adult stage',\n",
       "   'label': 'abnormal neuroanatomy | adult stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.029, 402.0, 0.088],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.07141778644953, 'y': 83.19326947531135}},\n",
       " {'data': {'id': 'photoreceptor cell R8 of pale ommatidium',\n",
       "   'label': 'photoreceptor cell R8 of pale ommatidium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 249.0, 0.055],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.87078928688964, 'y': 14.811722806573213}},\n",
       " {'data': {'id': 'photoreceptor cell R8 of yellow ommatidium',\n",
       "   'label': 'photoreceptor cell R8 of yellow ommatidium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 268.0, 0.059],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.684348988025734, 'y': 34.96592059886612}},\n",
       " {'data': {'id': 'bract',\n",
       "   'label': 'bract',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 269.0, 0.059],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.7640733170657463, 'y': 54.03569076620672}},\n",
       " {'data': {'id': 'labial sensillum',\n",
       "   'label': 'labial sensillum',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 106.0, 0.023],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.05405485378611, 'y': 11.621821362753671}},\n",
       " {'data': {'id': 'labial sensillum | somatic clone',\n",
       "   'label': 'labial sensillum | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 107.0, 0.024],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.41986054607922, 'y': 11.750611847942572}},\n",
       " {'data': {'id': 'visible | somatic clone',\n",
       "   'label': 'visible | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.075, 432.0, 0.095],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.87111027154, 'y': 76.68190508834375}},\n",
       " {'data': {'id': 'ENSG00000096384',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000096384',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 62.5}},\n",
       " {'data': {'id': 'FBgn0001233',\n",
       "   'label': 'FBgn0001233',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 62.625160799685354, 'y': 24.055602698394686}},\n",
       " {'data': {'id': 'wild-type',\n",
       "   'label': 'wild-type',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.017, 380.0, 0.084],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.85445067185243, 'y': 37.44349081583693}},\n",
       " {'data': {'id': 'GO0016006:Nebenkern',\n",
       "   'label': 'GO0016006:Nebenkern',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.032, 407.0, 0.089],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.96009761177663, 'y': 70.872774241265}},\n",
       " {'data': {'id': 'Nebenkern derivative',\n",
       "   'label': 'Nebenkern derivative',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 330.0, 0.073],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.575621532026798, 'y': 27.175773024389027}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm',\n",
       "   'label': 'abnormal locomotor rhythm',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 4.0, 0.001],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.37313127302578, 'y': 33.89149220727371}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Hsp83[e6A])',\n",
       "   'label': 'abnormal locomotor rhythm (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 90.0, 0.02],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.71938577606265, 'y': 92.81778560060731}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Hsp83[e6D])',\n",
       "   'label': 'abnormal locomotor rhythm (with Hsp83[e6D])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 91.0, 0.02],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.90288777610155, 'y': 17.041679073849348}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Hsp83[j5C2])',\n",
       "   'label': 'abnormal locomotor rhythm (with Hsp83[j5C2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 92.0, 0.02],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.777165265951619, 'y': 35.92536479583451}},\n",
       " {'data': {'id': 'embryonic abdominal segment | maternal effect (with Hsp83[e6D])',\n",
       "   'label': 'embryonic abdominal segment | maternal effect (with Hsp83[e6D])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 93.0, 0.02],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.0509004241609263, 'y': 7.247401784404717}},\n",
       " {'data': {'id': 'female fertile',\n",
       "   'label': 'female fertile',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.08, 436.0, 0.096],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.62475356889969, 'y': 57.86360098438315}},\n",
       " {'data': {'id': 'female semi-fertile',\n",
       "   'label': 'female semi-fertile',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 6.0, 0.001],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.713768599918508, 'y': 84.2052393414414}},\n",
       " {'data': {'id': 'primary spermatocyte',\n",
       "   'label': 'primary spermatocyte',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 286.0, 0.063],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 14.53271292195598, 'y': 3.292293409520386}},\n",
       " {'data': {'id': 'primary spermatocyte cyst',\n",
       "   'label': 'primary spermatocyte cyst',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 345.0, 0.076],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.120114255413605, 'y': 72.64263694314566}},\n",
       " {'data': {'id': 'spermatid',\n",
       "   'label': 'spermatid',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.052, 421.0, 0.093],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.987542049355262, 'y': 51.80704681483517}},\n",
       " {'data': {'id': 'spermatid axoneme',\n",
       "   'label': 'spermatid axoneme',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.011, 360.0, 0.079],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.84949770632632, 'y': 0.2683152717064652}},\n",
       " {'data': {'id': 'spermatocyte',\n",
       "   'label': 'spermatocyte',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.023, 386.0, 0.085],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.4541793000466, 'y': 83.37838562440776}},\n",
       " {'data': {'id': 'spermatozoon',\n",
       "   'label': 'spermatozoon',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.065, 426.0, 0.094],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.67835036255421, 'y': 9.571183670253891}},\n",
       " {'data': {'id': 'abnormal cell polarity (with Hsp83[P582])',\n",
       "   'label': 'abnormal cell polarity (with Hsp83[P582])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 139.0, 0.031],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.951514438753572, 'y': 28.351086776566426}},\n",
       " {'data': {'id': 'lethal - all die before end of larval stage | recessive',\n",
       "   'label': 'lethal - all die before end of larval stage | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.024, 391.0, 0.086],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.5155277017182542, 'y': 47.69522821706974}},\n",
       " {'data': {'id': 'neuroblast (with Hsp83[P582])',\n",
       "   'label': 'neuroblast (with Hsp83[P582])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 94.0, 0.021],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.576113049906562, 'y': 62.982151603751376}},\n",
       " {'data': {'id': 'partially lethal (with Hsp83[6-55])',\n",
       "   'label': 'partially lethal (with Hsp83[6-55])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 114.0, 0.025],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.5657853491432, 'y': 98.80119030596947}},\n",
       " {'data': {'id': 'viable (with Hsp83[e1D])',\n",
       "   'label': 'viable (with Hsp83[e1D])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 116.0, 0.025],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.3218619309166, 'y': 88.26580230484613}},\n",
       " {'data': {'id': 'viable (with Hsp83[e3A])',\n",
       "   'label': 'viable (with Hsp83[e3A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 151.0, 0.033],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.446535948361864, 'y': 34.890196799338014}},\n",
       " {'data': {'id': 'dorsal appendage (with Hsp83[e6A])',\n",
       "   'label': 'dorsal appendage (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 117.0, 0.026],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.68119458206878, 'y': 73.73899337342364}},\n",
       " {'data': {'id': 'egg (with Hsp83[e6A])',\n",
       "   'label': 'egg (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 136.0, 0.03],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.341463838887396, 'y': 94.73406479130036}},\n",
       " {'data': {'id': 'egg chamber (with Hsp83[e6A])',\n",
       "   'label': 'egg chamber (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 137.0, 0.03],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.3876979106477139, 'y': 70.2898238139873}},\n",
       " {'data': {'id': 'nurse cell (with Hsp83[e6A])',\n",
       "   'label': 'nurse cell (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 138.0, 0.03],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.264256074614542, 'y': 1.730137021410194}},\n",
       " {'data': {'id': 'eye disc | somatic clone | third instar larval stage',\n",
       "   'label': 'eye disc | somatic clone | third instar larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.01, 356.0, 0.078],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.791158026584966, 'y': 76.71779117498978}},\n",
       " {'data': {'id': 'increased occurrence of cell division | pupal stage | somatic clone',\n",
       "   'label': 'increased occurrence of cell division | pupal stage | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 229.0, 0.05],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.03664148745817, 'y': 45.92979503021408}},\n",
       " {'data': {'id': 'increased occurrence of cell division | somatic clone | third instar larval stage',\n",
       "   'label': 'increased occurrence of cell division | somatic clone | third instar larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 264.0, 0.058],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.83278779011647, 'y': 76.28935491588038}},\n",
       " {'data': {'id': 'lethal (with Hsp83[P582])',\n",
       "   'label': 'lethal (with Hsp83[P582])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 89.0, 0.02],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.347190876605374, 'y': 40.983826085192256}},\n",
       " {'data': {'id': 'lethal (with Hsp83[j5C2])',\n",
       "   'label': 'lethal (with Hsp83[j5C2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 98.0, 0.022],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.79584193325371, 'y': 38.508157868038694}},\n",
       " {'data': {'id': 'lethal - all die before end of pupal stage',\n",
       "   'label': 'lethal - all die before end of pupal stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.072, 431.0, 0.095],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.28455346038248, 'y': 90.74165128932113}},\n",
       " {'data': {'id': 'male sterile (with Hsp83[e1D])',\n",
       "   'label': 'male sterile (with Hsp83[e1D])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 97.0, 0.021],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.3680789640326, 'y': 78.98013140527125}},\n",
       " {'data': {'id': 'male sterile (with Hsp83[e3A])',\n",
       "   'label': 'male sterile (with Hsp83[e3A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 96.0, 0.021],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.76359011848363, 'y': 38.4865934572703}},\n",
       " {'data': {'id': 'male sterile (with Hsp83[e4A])',\n",
       "   'label': 'male sterile (with Hsp83[e4A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 87.0, 0.019],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.349188067453778, 'y': 33.741313394122784}},\n",
       " {'data': {'id': 'male sterile (with Hsp83[e6A])',\n",
       "   'label': 'male sterile (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 81.0, 0.018],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.60255340104742, 'y': 40.345124423699495}},\n",
       " {'data': {'id': 'male sterile (with Hsp83[e6D])',\n",
       "   'label': 'male sterile (with Hsp83[e6D])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 82.0, 0.018],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.18851154684911, 'y': 53.71277326259023}},\n",
       " {'data': {'id': 'partially lethal (with Hsp83[13F3])',\n",
       "   'label': 'partially lethal (with Hsp83[13F3])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 83.0, 0.018],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.7414029789584626, 'y': 62.9740998861557}},\n",
       " {'data': {'id': 'partially lethal (with Hsp83[19F2])',\n",
       "   'label': 'partially lethal (with Hsp83[19F2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 84.0, 0.018],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.13783653946912, 'y': 50.34605763880388}},\n",
       " {'data': {'id': 'GO0005813:centrosome',\n",
       "   'label': 'GO0005813:centrosome',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.012, 362.0, 0.08],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.26657470186151, 'y': 7.3492540127855115}},\n",
       " {'data': {'id': 'GO0072686:mitotic spindle | larval stage',\n",
       "   'label': 'GO0072686:mitotic spindle | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 234.0, 0.051],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.63421076019, 'y': 86.24946298010302}},\n",
       " {'data': {'id': 'abnormal mitotic cell cycle | larval stage',\n",
       "   'label': 'abnormal mitotic cell cycle | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.009, 351.0, 0.077],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.529603524686446, 'y': 13.12721710250917}},\n",
       " {'data': {'id': 'anterior dorsocentral bristle',\n",
       "   'label': 'anterior dorsocentral bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 331.0, 0.073],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.39062983056013, 'y': 83.24742782833084}},\n",
       " {'data': {'id': 'embryonic/larval brain | larval stage',\n",
       "   'label': 'embryonic/larval brain | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.025, 394.0, 0.087],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.86600788928318, 'y': 7.968946168876901}},\n",
       " {'data': {'id': 'humeral bristle',\n",
       "   'label': 'humeral bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 337.0, 0.074],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.3241344875299, 'y': 56.729512596787856}},\n",
       " {'data': {'id': 'mesothoracic bristle | increased number',\n",
       "   'label': 'mesothoracic bristle | increased number',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 85.0, 0.019],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.16254629423192, 'y': 68.13231740508783}},\n",
       " {'data': {'id': 'metathoracic bristle | increased number',\n",
       "   'label': 'metathoracic bristle | increased number',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 77.0, 0.017],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.270025804955694, 'y': 72.56458486456275}},\n",
       " {'data': {'id': 'posterior dorsocentral bristle',\n",
       "   'label': 'posterior dorsocentral bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 309.0, 0.068],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.172000117491023, 'y': 29.427176946981547}},\n",
       " {'data': {'id': 'scutellar bristle | increased number',\n",
       "   'label': 'scutellar bristle | increased number',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 318.0, 0.07],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.15024282927139, 'y': 64.1126488246114}},\n",
       " {'data': {'id': 'viable (with Hsp83[e6A])',\n",
       "   'label': 'viable (with Hsp83[e6A])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 88.0, 0.019],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.19222250055826, 'y': 93.0024544433556}},\n",
       " {'data': {'id': 'viable (with Hsp83[e6D])',\n",
       "   'label': 'viable (with Hsp83[e6D])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 95.0, 0.021],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.162303721442761, 'y': 73.13281336293657}},\n",
       " {'data': {'id': 'GO0030017:sarcomere, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'label': 'GO0030017:sarcomere, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 301.0, 0.066],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.644360825322664, 'y': 65.64252933057382}},\n",
       " {'data': {'id': 'GO0030018:Z disc, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'label': 'GO0030018:Z disc, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.075, 433.0, 0.095],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.81681328645057, 'y': 14.720704936651273}},\n",
       " {'data': {'id': 'embryonic/larval somatic muscle, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'label': 'embryonic/larval somatic muscle, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.024, 390.0, 0.086],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.9052905110389275, 'y': 97.01753271596492}},\n",
       " {'data': {'id': 'lethal | male, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'lethal | male, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 271.0, 0.06],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.57558488399507, 'y': 90.85273375888173}},\n",
       " {'data': {'id': 'lethal, with Scer\\\\GAL4[Act5C.PU]',\n",
       "   'label': 'lethal, with Scer\\\\GAL4[Act5C.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.067, 429.0, 0.094],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.539973973823859, 'y': 38.28723451462898}},\n",
       " {'data': {'id': 'lethal, with Scer\\\\GAL4[GMR.PF], Scer\\\\GAL4[ey.PH]',\n",
       "   'label': 'lethal, with Scer\\\\GAL4[GMR.PF], Scer\\\\GAL4[ey.PH]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 315.0, 0.069],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.091778962084128, 'y': 36.9692078266919}},\n",
       " {'data': {'id': 'partially lethal - majority die | female, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'partially lethal - majority die | female, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 225.0, 0.049],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.6875927016199, 'y': 60.14820620375776}},\n",
       " {'data': {'id': 'partially lethal, with Scer\\\\GAL4[tin.CΔ4]',\n",
       "   'label': 'partially lethal, with Scer\\\\GAL4[tin.CΔ4]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.065, 427.0, 0.094],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.12638300231315, 'y': 5.326069159212709}},\n",
       " {'data': {'id': 'some die during pupal stage, with Scer\\\\GAL4[tin.CΔ4]',\n",
       "   'label': 'some die during pupal stage, with Scer\\\\GAL4[tin.CΔ4]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 327.0, 0.072],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.9918347042297824, 'y': 96.57325715357028}},\n",
       " {'data': {'id': 'visible, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'visible, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.01, 358.0, 0.079],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.11168950385635, 'y': 64.63529239877256}},\n",
       " {'data': {'id': 'wing, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'wing, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.013, 365.0, 0.08],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.141675003602652, 'y': 14.307160045648693}},\n",
       " {'data': {'id': 'photoreceptor cell R7',\n",
       "   'label': 'photoreceptor cell R7',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.027, 399.0, 0.088],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.532001735730025, 'y': 60.66059151404075}},\n",
       " {'data': {'id': 'secondary pigment cell',\n",
       "   'label': 'secondary pigment cell',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 340.0, 0.075],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.563664918212503, 'y': 54.02975889115904}},\n",
       " {'data': {'id': 'increased cell death | third instar larval stage, with Scer\\\\GAL4[en-e16E]',\n",
       "   'label': 'increased cell death | third instar larval stage, with Scer\\\\GAL4[en-e16E]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 328.0, 0.072],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.767793588672767, 'y': 66.62337322252527}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[da.G32]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[da.G32]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.016, 377.0, 0.083],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.89743599103357, 'y': 13.45239157149093}},\n",
       " {'data': {'id': 'wing disc | third instar larval stage, with Scer\\\\GAL4[en-e16E]',\n",
       "   'label': 'wing disc | third instar larval stage, with Scer\\\\GAL4[en-e16E]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 348.0, 0.076],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.46722790089719, 'y': 98.76995529495332}},\n",
       " {'data': {'id': 'abnormal developmental rate | larval stage, with Scer\\\\GAL4[da.G32]',\n",
       "   'label': 'abnormal developmental rate | larval stage, with Scer\\\\GAL4[da.G32]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 272.0, 0.06],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.97022069511411, 'y': 20.649816566530443}},\n",
       " {'data': {'id': 'lethal - all die during third instar larval stage, with Scer\\\\GAL4[da.G32]',\n",
       "   'label': 'lethal - all die during third instar larval stage, with Scer\\\\GAL4[da.G32]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 164.0, 0.036],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.51957146697595, 'y': 81.23790123557286}},\n",
       " {'data': {'id': 'GO0048786:presynaptic active zone | decreased number | third instar larval stage, with Scer\\\\GAL4[Toll-6-D42]',\n",
       "   'label': 'GO0048786:presynaptic active zone | decreased number | third instar larval stage, with Scer\\\\GAL4[Toll-6-D42]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 275.0, 0.06],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.42970025507523, 'y': 80.58281783827647}},\n",
       " {'data': {'id': 'embryonic/larval neuromuscular junction | third instar larval stage, with Scer\\\\GAL4[Toll-6-D42]',\n",
       "   'label': 'embryonic/larval neuromuscular junction | third instar larval stage, with Scer\\\\GAL4[Toll-6-D42]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 339.0, 0.075],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.150888320962398, 'y': 45.41238366197932}},\n",
       " {'data': {'id': 'increased occurrence of cell division | pupal stage, with Scer\\\\GAL4[GMR.PU]',\n",
       "   'label': 'increased occurrence of cell division | pupal stage, with Scer\\\\GAL4[GMR.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 203.0, 0.045],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.08775125901452, 'y': 83.02628380202225}},\n",
       " {'data': {'id': 'increased occurrence of cell division | pupal stage, with Scer\\\\GAL4[ap-md544]',\n",
       "   'label': 'increased occurrence of cell division | pupal stage, with Scer\\\\GAL4[ap-md544]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 172.0, 0.038],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.12910051597556427, 'y': 91.2476270799249}},\n",
       " {'data': {'id': 'wing | dorsal, with Scer\\\\GAL4[ap-md544]',\n",
       "   'label': 'wing | dorsal, with Scer\\\\GAL4[ap-md544]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 198.0, 0.044],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.979999153417022, 'y': 10.017102068687933}},\n",
       " {'data': {'id': 'abnormal cell polarity (with Hsp83[13F3])',\n",
       "   'label': 'abnormal cell polarity (with Hsp83[13F3])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 188.0, 0.041],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.46651972078611, 'y': 31.00681949613283}},\n",
       " {'data': {'id': 'lethal (with Hsp83[6-55])',\n",
       "   'label': 'lethal (with Hsp83[6-55])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 185.0, 0.041],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.410229700447147, 'y': 57.14204718783833}},\n",
       " {'data': {'id': 'neuroblast (with Hsp83[13F3])',\n",
       "   'label': 'neuroblast (with Hsp83[13F3])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 181.0, 0.04],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.58391042943409, 'y': 46.82894089228205}},\n",
       " {'data': {'id': 'short lived',\n",
       "   'label': 'short lived',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.082, 437.0, 0.096],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.25022685490575, 'y': 69.19201552593789}},\n",
       " {'data': {'id': 'chaeta',\n",
       "   'label': 'chaeta',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.02, 384.0, 0.084],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.42594298028983, 'y': 71.28552447076781}},\n",
       " {'data': {'id': 'male sterile (with Hsp83[6-55])',\n",
       "   'label': 'male sterile (with Hsp83[6-55])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 177.0, 0.039],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.452268838439094, 'y': 43.167843784646344}},\n",
       " {'data': {'id': 'viable (with Hsp83[13F3])',\n",
       "   'label': 'viable (with Hsp83[13F3])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 178.0, 0.039],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.9222921262129273, 'y': 4.548658704593356}},\n",
       " {'data': {'id': 'viable (with Hsp83[19F2])',\n",
       "   'label': 'viable (with Hsp83[19F2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 179.0, 0.039],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.443086545708177, 'y': 15.34869866986841}},\n",
       " {'data': {'id': 'viable (with Hsp83[9J1])',\n",
       "   'label': 'viable (with Hsp83[9J1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 183.0, 0.04],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.84744133820841, 'y': 23.408251358885945}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Hsp83[08445])',\n",
       "   'label': 'abnormal locomotor rhythm (with Hsp83[08445])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 187.0, 0.041],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.1818223385307243, 'y': 30.640881170506738}},\n",
       " {'data': {'id': 'dorsal appendage (with Hsp83[19F2])',\n",
       "   'label': 'dorsal appendage (with Hsp83[19F2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 154.0, 0.034],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.88152971657223, 'y': 63.300003215725134}},\n",
       " {'data': {'id': 'egg (with Hsp83[19F2])',\n",
       "   'label': 'egg (with Hsp83[19F2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 173.0, 0.038],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.80470463438458, 'y': 1.4449320636202967}},\n",
       " {'data': {'id': 'egg chamber (with Hsp83[19F2])',\n",
       "   'label': 'egg chamber (with Hsp83[19F2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 176.0, 0.039],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.833758172163337, 'y': 48.873039667701924}},\n",
       " {'data': {'id': 'nurse cell (with Hsp83[19F2])',\n",
       "   'label': 'nurse cell (with Hsp83[19F2])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 184.0, 0.04],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.645758680384239, 'y': 49.81805834477087}},\n",
       " {'data': {'id': 'embryonic abdominal segment | maternal effect (with Hsp83[08445])',\n",
       "   'label': 'embryonic abdominal segment | maternal effect (with Hsp83[08445])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 180.0, 0.04],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.744555317707277, 'y': 44.177270917352985}},\n",
       " {'data': {'id': 'ENSG00000106546',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000106546',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 75.0}},\n",
       " {'data': {'id': 'ENSG00000110711',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000110711',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 87.5}},\n",
       " {'data': {'id': 'FBgn0030345',\n",
       "   'label': 'FBgn0030345',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 21.457655024427172, 'y': 74.13376719598664}},\n",
       " {'data': {'id': 'ENSG00000110958',\n",
       "   'indent': 'gene',\n",
       "   'label': 'ENSG00000110958',\n",
       "   'size': 4,\n",
       "   'fontsize': '1.5px'},\n",
       "  'classes': 'blue',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 50, 'y': 100.0}},\n",
       " {'data': {'id': 'FBgn0037728',\n",
       "   'label': 'FBgn0037728',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 21.761061843778254, 'y': 67.30682742688246}},\n",
       " {'data': {'id': 'FBgn0264075',\n",
       "   'label': 'FBgn0264075',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 71.25870837613617, 'y': 55.83045550786758}},\n",
       " {'data': {'id': 'GO0030424:axon',\n",
       "   'label': 'GO0030424:axon',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.014, 371.0, 0.082],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.16669674656391, 'y': 12.783934513009099}},\n",
       " {'data': {'id': 'VUM neuron',\n",
       "   'label': 'VUM neuron',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 284.0, 0.062],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.118162334896976, 'y': 66.0613082844675}},\n",
       " {'data': {'id': 'abnormal cell migration | oogenesis stage S10 | somatic clone',\n",
       "   'label': 'abnormal cell migration | oogenesis stage S10 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 206.0, 0.045],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.41186522459289, 'y': 92.11455280990377}},\n",
       " {'data': {'id': 'abnormal cell migration | oogenesis stage S9 | somatic clone',\n",
       "   'label': 'abnormal cell migration | oogenesis stage S9 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 209.0, 0.046],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.433638077906489, 'y': 63.23635611921278}},\n",
       " {'data': {'id': 'antenna | somatic clone',\n",
       "   'label': 'antenna | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 313.0, 0.069],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.71173634636774, 'y': 23.89855339431499}},\n",
       " {'data': {'id': 'arista | somatic clone',\n",
       "   'label': 'arista | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 281.0, 0.062],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.6465979662339, 'y': 53.84057276752835}},\n",
       " {'data': {'id': 'basal cylinder | somatic clone',\n",
       "   'label': 'basal cylinder | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 199.0, 0.044],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.93371952637759, 'y': 54.563954528971905}},\n",
       " {'data': {'id': 'border follicle cell | oogenesis stage S10 | somatic clone',\n",
       "   'label': 'border follicle cell | oogenesis stage S10 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 246.0, 0.054],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.229907383808619, 'y': 28.797146365450033}},\n",
       " {'data': {'id': 'border follicle cell | oogenesis stage S9 | somatic clone',\n",
       "   'label': 'border follicle cell | oogenesis stage S9 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 193.0, 0.042],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.43564840749887, 'y': 27.110028905850537}},\n",
       " {'data': {'id': 'embryonic/larval tracheal system',\n",
       "   'label': 'embryonic/larval tracheal system',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.043, 417.0, 0.092],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.53108691070865, 'y': 83.0194351140004}},\n",
       " {'data': {'id': 'leg | distal | somatic clone',\n",
       "   'label': 'leg | distal | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 191.0, 0.042],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.511130173320989, 'y': 29.339826706646367}},\n",
       " {'data': {'id': 'macrochaeta | somatic clone',\n",
       "   'label': 'macrochaeta | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.015, 373.0, 0.082],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.9551907111070079, 'y': 30.796968793567046}},\n",
       " {'data': {'id': 'maxillary palpus | somatic clone',\n",
       "   'label': 'maxillary palpus | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 290.0, 0.064],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.95675642125597, 'y': 46.27500926163487}},\n",
       " {'data': {'id': 'midline glial cell',\n",
       "   'label': 'midline glial cell',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.012, 363.0, 0.08],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.929374758343318, 'y': 28.435327361926298}},\n",
       " {'data': {'id': 'neuroblast MNB',\n",
       "   'label': 'neuroblast MNB',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 253.0, 0.056],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.672880335718197, 'y': 50.83733388871027}},\n",
       " {'data': {'id': 'pretarsus | somatic clone',\n",
       "   'label': 'pretarsus | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 276.0, 0.061],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.131483534487469, 'y': 18.915212614015108}},\n",
       " {'data': {'id': 'tarsal segment 4 | somatic clone',\n",
       "   'label': 'tarsal segment 4 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 266.0, 0.058],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.8035316545096434, 'y': 46.54292950406833}},\n",
       " {'data': {'id': 'tarsal segment | somatic clone',\n",
       "   'label': 'tarsal segment | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 314.0, 0.069],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.89251920143258, 'y': 91.13700111677335}},\n",
       " {'data': {'id': 'third segment of antenna | distal | somatic clone',\n",
       "   'label': 'third segment of antenna | distal | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 160.0, 0.035],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.530643698414192, 'y': 56.90163115923244}},\n",
       " {'data': {'id': 'unguis | ectopic | somatic clone',\n",
       "   'label': 'unguis | ectopic | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 201.0, 0.044],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 14.013459219861435, 'y': 23.646753142462085}},\n",
       " {'data': {'id': 'visible | recessive | somatic clone',\n",
       "   'label': 'visible | recessive | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.038, 412.0, 0.091],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.5648642770936, 'y': 36.90291977865987}},\n",
       " {'data': {'id': 'wing | somatic clone',\n",
       "   'label': 'wing | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.051, 420.0, 0.092],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.5643772328598757, 'y': 81.21664913492582}},\n",
       " {'data': {'id': 'dorsal branch primordium',\n",
       "   'label': 'dorsal branch primordium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 332.0, 0.073],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.60241912216321, 'y': 89.7971652040404}},\n",
       " {'data': {'id': 'dorsal trunk primordium',\n",
       "   'label': 'dorsal trunk primordium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.011, 359.0, 0.079],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.40917399490182, 'y': 53.86145001266115}},\n",
       " {'data': {'id': 'ganglionic branch primordium',\n",
       "   'label': 'ganglionic branch primordium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 333.0, 0.073],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.57961055081618, 'y': 97.57107208377847}},\n",
       " {'data': {'id': 'presumptive embryonic/larval tracheal system',\n",
       "   'label': 'presumptive embryonic/larval tracheal system',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.017, 378.0, 0.083],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 14.170060203317842, 'y': 68.31310322156105}},\n",
       " {'data': {'id': 'transverse connective primordium',\n",
       "   'label': 'transverse connective primordium',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 251.0, 0.055],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.18955490366504, 'y': 7.5417227218764875}},\n",
       " {'data': {'id': 'tarsal segment 2 | somatic clone',\n",
       "   'label': 'tarsal segment 2 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 242.0, 0.053],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.852557332733839, 'y': 30.510313270192878}},\n",
       " {'data': {'id': 'tarsal segment 3 | somatic clone',\n",
       "   'label': 'tarsal segment 3 | somatic clone',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 241.0, 0.053],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.50325765161637, 'y': 72.39723027553654}},\n",
       " {'data': {'id': 'abnormal cell migration',\n",
       "   'label': 'abnormal cell migration',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 329.0, 0.072],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.6164812887419395, 'y': 81.42894238642344}},\n",
       " {'data': {'id': 'exit glial cell',\n",
       "   'label': 'exit glial cell',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 254.0, 0.056],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.273449001141092, 'y': 92.82635647093383}},\n",
       " {'data': {'id': 'larval intersegmental nerve',\n",
       "   'label': 'larval intersegmental nerve',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.015, 372.0, 0.082],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.11188932120511, 'y': 88.63081837595927}},\n",
       " {'data': {'id': 'larval longitudinal connective',\n",
       "   'label': 'larval longitudinal connective',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.032, 408.0, 0.09],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.430759453711339, 'y': 68.62716783018088}},\n",
       " {'data': {'id': 'larval segmental nerve',\n",
       "   'label': 'larval segmental nerve',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 305.0, 0.067],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.64673418998801, 'y': 21.519722176121057}},\n",
       " {'data': {'id': 'subperineurial glial cell',\n",
       "   'label': 'subperineurial glial cell',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 252.0, 0.055],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.10310203746332, 'y': 45.10528492380702}},\n",
       " {'data': {'id': 'abnormal cell migration, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'label': 'abnormal cell migration, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 169.0, 0.037],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.48749541629934, 'y': 44.76829450449999}},\n",
       " {'data': {'id': 'abnormal neuroanatomy, with Scer\\\\GAL4[NP6099]',\n",
       "   'label': 'abnormal neuroanatomy, with Scer\\\\GAL4[NP6099]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 218.0, 0.048],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.675730954593055, 'y': 9.319780503199349}},\n",
       " {'data': {'id': 'abnormal neuroanatomy, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'label': 'abnormal neuroanatomy, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 194.0, 0.043],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.28011874009272, 'y': 76.53428472024478}},\n",
       " {'data': {'id': 'embryonic/larval brain, with Scer\\\\GAL4[NP6099]',\n",
       "   'label': 'embryonic/larval brain, with Scer\\\\GAL4[NP6099]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 171.0, 0.038],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.855210807700561, 'y': 71.74230357925327}},\n",
       " {'data': {'id': 'embryonic/larval tracheal system, with Scer\\\\GAL4[btl.PS]',\n",
       "   'label': 'embryonic/larval tracheal system, with Scer\\\\GAL4[btl.PS]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 323.0, 0.071],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.02973933548137, 'y': 88.33199132404047}},\n",
       " {'data': {'id': 'exit glial cell, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'label': 'exit glial cell, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 175.0, 0.038],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.57534558354395, 'y': 85.16050670090051}},\n",
       " {'data': {'id': 'glial cell, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'label': 'glial cell, with Scer\\\\GAL4[loco.1.3]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 156.0, 0.034],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.23727001453469, 'y': 59.276166919286474}},\n",
       " {'data': {'id': 'lamina forming neuroblast, with Scer\\\\GAL4[NP6099]',\n",
       "   'label': 'lamina forming neuroblast, with Scer\\\\GAL4[NP6099]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 197.0, 0.043],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.731595583971362, 'y': 47.60657359191243}},\n",
       " {'data': {'id': 'abnormal neuroanatomy, with Scer\\\\GAL4[btl.PS]',\n",
       "   'label': 'abnormal neuroanatomy, with Scer\\\\GAL4[btl.PS]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 167.0, 0.037],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.1917123564238, 'y': 26.76767102416878}},\n",
       " {'data': {'id': 'abnormal neuroanatomy, with Scer\\\\GAL4[sim.P3.7]',\n",
       "   'label': 'abnormal neuroanatomy, with Scer\\\\GAL4[sim.P3.7]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 270.0, 0.059],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.72935839471779, 'y': 6.7207906245969395}},\n",
       " {'data': {'id': 'larval longitudinal connective, with Scer\\\\GAL4[sim.P3.7]',\n",
       "   'label': 'larval longitudinal connective, with Scer\\\\GAL4[sim.P3.7]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 226.0, 0.05],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.67793145397941, 'y': 67.41615557504284}},\n",
       " {'data': {'id': 'midline glial cell, with Scer\\\\GAL4[btl.PS]',\n",
       "   'label': 'midline glial cell, with Scer\\\\GAL4[btl.PS]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 155.0, 0.034],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.04564913287454, 'y': 57.31375631154725}},\n",
       " {'data': {'id': 'symmetrical commissure, with Scer\\\\GAL4[sim.P3.7]',\n",
       "   'label': 'symmetrical commissure, with Scer\\\\GAL4[sim.P3.7]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 222.0, 0.049],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.95391614302778, 'y': 86.52743494538805}},\n",
       " {'data': {'id': 'tracheal system, with Scer\\\\GAL4[btl.PS]',\n",
       "   'label': 'tracheal system, with Scer\\\\GAL4[btl.PS]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 174.0, 0.038],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.5523748647947646, 'y': 80.44637636515117}},\n",
       " {'data': {'id': 'peripheral nervous system',\n",
       "   'label': 'peripheral nervous system',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 341.0, 0.075],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.6332079604815, 'y': 58.42096479043239}},\n",
       " {'data': {'id': 'leg joint, with Scer\\\\GAL4[GMR13D07]',\n",
       "   'label': 'leg joint, with Scer\\\\GAL4[GMR13D07]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 157.0, 0.035],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.375448940597794, 'y': 53.47105086844465}},\n",
       " {'data': {'id': 'tarsal segment, with Scer\\\\GAL4[GMR13D07]',\n",
       "   'label': 'tarsal segment, with Scer\\\\GAL4[GMR13D07]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 158.0, 0.035],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.049973094060895, 'y': 25.92205048916687}},\n",
       " {'data': {'id': 'adult midgut, with Scer\\\\GAL4[NP0001]',\n",
       "   'label': 'adult midgut, with Scer\\\\GAL4[NP0001]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 310.0, 0.068],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.8503433067269017, 'y': 76.81234604583572}},\n",
       " {'data': {'id': 'increased occurrence of cell division | adult stage, with Scer\\\\GAL4[NP0001]',\n",
       "   'label': 'increased occurrence of cell division | adult stage, with Scer\\\\GAL4[NP0001]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 346.0, 0.076],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.05748892046415, 'y': 18.650054273897577}},\n",
       " {'data': {'id': 'short lived, with Scer\\\\GAL4[NP0001]',\n",
       "   'label': 'short lived, with Scer\\\\GAL4[NP0001]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 300.0, 0.066],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.4128320580602, 'y': 14.449107359958735}},\n",
       " {'data': {'id': 'lethal | male limited | cold sensitive, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'lethal | male limited | cold sensitive, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.025, 392.0, 0.086],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.06295748956089, 'y': 90.25252704169337}},\n",
       " {'data': {'id': 'lethal | male limited, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'lethal | male limited, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.03, 403.0, 0.089],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.71018788430426, 'y': 1.7214088580355669}},\n",
       " {'data': {'id': 'visible | female limited | cold sensitive, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'visible | female limited | cold sensitive, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.029, 401.0, 0.088],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.06702902001274, 'y': 19.371403728287582}},\n",
       " {'data': {'id': 'visible | female limited, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'visible | female limited, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.035, 409.0, 0.09],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.236125939479434, 'y': 39.317974121560916}},\n",
       " {'data': {'id': 'wing | female limited | cold sensitive, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'wing | female limited | cold sensitive, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.025, 393.0, 0.086],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.927132140138259, 'y': 35.59640220832859}},\n",
       " {'data': {'id': 'wing | female limited, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'label': 'wing | female limited, with Scer\\\\GAL4[Bx-MS1096]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.03, 404.0, 0.089],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.66902329162113, 'y': 9.558018231856346}},\n",
       " {'data': {'id': 'leg, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'label': 'leg, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 239.0, 0.053],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.5587192560600585, 'y': 81.25395408837845}},\n",
       " {'data': {'id': 'tarsal segment 4, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'label': 'tarsal segment 4, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 162.0, 0.036],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.485187364464501, 'y': 40.078399304288205}},\n",
       " {'data': {'id': 'tarsal segment 5, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'label': 'tarsal segment 5, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 163.0, 0.036],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.57546476753542, 'y': 48.68358645684442}},\n",
       " {'data': {'id': 'antenna, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'label': 'antenna, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 208.0, 0.046],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.562343982197095, 'y': 44.72360144119529}},\n",
       " {'data': {'id': 'leg | ectopic, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'label': 'leg | ectopic, with Scer\\\\GAL4[Dll-em212]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 161.0, 0.035],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.05587806792076, 'y': 97.7518932505081}},\n",
       " {'data': {'id': 'presumptive embryonic/larval central nervous system',\n",
       "   'label': 'presumptive embryonic/larval central nervous system',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.036, 411.0, 0.09],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.815331284006167, 'y': 78.84725336743995}},\n",
       " {'data': {'id': 'symmetrical commissure',\n",
       "   'label': 'symmetrical commissure',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.025, 395.0, 0.087],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.193026980420443, 'y': 73.5541951334172}},\n",
       " {'data': {'id': 'FBgn0002723',\n",
       "   'label': 'FBgn0002723',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 76.5319763191258, 'y': 14.052616946164342}},\n",
       " {'data': {'id': 'abdominal sternite bristle',\n",
       "   'label': 'abdominal sternite bristle',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 317.0, 0.07],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.310967420906996, 'y': 94.68412007862571}},\n",
       " {'data': {'id': 'male genitalia',\n",
       "   'label': 'male genitalia',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.015, 374.0, 0.082],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.23155241518253, 'y': 66.29832018316131}},\n",
       " {'data': {'id': 'chemical resistant',\n",
       "   'label': 'chemical resistant',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.026, 397.0, 0.087],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.4039535711322682, 'y': 14.869663126318077}},\n",
       " {'data': {'id': 'chemical resistant (with Metatarsi-irregular[1])',\n",
       "   'label': 'chemical resistant (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 170.0, 0.037],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.119233644418688, 'y': 69.33487349232034}},\n",
       " {'data': {'id': 'chemical resistant | dominant',\n",
       "   'label': 'chemical resistant | dominant',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.013, 366.0, 0.08],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.204249536546603, 'y': 20.53195628634321}},\n",
       " {'data': {'id': 'chemical resistant | semidominant',\n",
       "   'label': 'chemical resistant | semidominant',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 243.0, 0.053],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.82731243577086, 'y': 38.89387422438554}},\n",
       " {'data': {'id': 'eye (with Metatarsi-irregular[1])',\n",
       "   'label': 'eye (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 71.0, 0.016],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.164210763875637, 'y': 94.94530120096287}},\n",
       " {'data': {'id': 'leg sensillum (with Metatarsi-irregular[1])',\n",
       "   'label': 'leg sensillum (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 72.0, 0.016],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.591215513721797, 'y': 4.039026182050687}},\n",
       " {'data': {'id': 'tarsal segment 1 & joint (with Metatarsi-irregular[1])',\n",
       "   'label': 'tarsal segment 1 & joint (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 73.0, 0.016],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.98690788639865, 'y': 92.05598402775841}},\n",
       " {'data': {'id': 'tarsal segment 2 & joint (with Metatarsi-irregular[1])',\n",
       "   'label': 'tarsal segment 2 & joint (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 74.0, 0.016],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.84485309725505, 'y': 9.367600941346844}},\n",
       " {'data': {'id': 'visible (with Metatarsi-irregular[1])',\n",
       "   'label': 'visible (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 76.0, 0.017],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.2912004989566, 'y': 9.352720272985737}},\n",
       " {'data': {'id': 'visible | dominant (with Metatarsi-irregular[1])',\n",
       "   'label': 'visible | dominant (with Metatarsi-irregular[1])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 59.0, 0.013],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.2408262011209805, 'y': 46.2532891806903}},\n",
       " {'data': {'id': 'adult optic lobe | prepupal stage',\n",
       "   'label': 'adult optic lobe | prepupal stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 61.0, 0.013],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.29516093618028, 'y': 33.21874962018519}},\n",
       " {'data': {'id': 'chemical resistant | larval stage',\n",
       "   'label': 'chemical resistant | larval stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 2.0, 0.0],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.227246195656351, 'y': 34.83950886912853}},\n",
       " {'data': {'id': 'decreased fecundity | female',\n",
       "   'label': 'decreased fecundity | female',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 5.0, 0.001],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.40986265965222, 'y': 45.33979947347299}},\n",
       " {'data': {'id': 'decreased occurrence of cell division | prepupal stage',\n",
       "   'label': 'decreased occurrence of cell division | prepupal stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 60.0, 0.013],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.45954940657911, 'y': 36.165370753862256}},\n",
       " {'data': {'id': 'lobula',\n",
       "   'label': 'lobula',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 312.0, 0.069],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.88566659081854, 'y': 86.95046967394092}},\n",
       " {'data': {'id': 'lobula plate',\n",
       "   'label': 'lobula plate',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 295.0, 0.065],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.50817367766315, 'y': 37.95453370783994}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S10',\n",
       "   'label': 'oocyte | oogenesis stage S10',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 240.0, 0.053],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.46688980298316, 'y': 60.276692263426554}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S11',\n",
       "   'label': 'oocyte | oogenesis stage S11',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 224.0, 0.049],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.21204246216783, 'y': 7.328792979964938}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S12',\n",
       "   'label': 'oocyte | oogenesis stage S12',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 221.0, 0.049],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.238218389277769, 'y': 28.703013016860456}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S13',\n",
       "   'label': 'oocyte | oogenesis stage S13',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 232.0, 0.051],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.029891045337987, 'y': 57.04261696858315}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S14',\n",
       "   'label': 'oocyte | oogenesis stage S14',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 289.0, 0.064],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.091224847246972, 'y': 2.8792460309135293}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S8',\n",
       "   'label': 'oocyte | oogenesis stage S8',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 237.0, 0.052],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.171089420313667, 'y': 32.922969438615425}},\n",
       " {'data': {'id': 'oocyte | oogenesis stage S9',\n",
       "   'label': 'oocyte | oogenesis stage S9',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 267.0, 0.059],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.432914033956253, 'y': 64.44427570938971}},\n",
       " {'data': {'id': 'ovary',\n",
       "   'label': 'ovary',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.06, 424.0, 0.093],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.608363580769303, 'y': 99.78611562368722}},\n",
       " {'data': {'id': 'photoreceptor cell R7 | prepupal stage',\n",
       "   'label': 'photoreceptor cell R7 | prepupal stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 70.0, 0.015],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.2293534520444482, 'y': 73.55918589449757}},\n",
       " {'data': {'id': 'photoreceptor cell R8 | prepupal stage',\n",
       "   'label': 'photoreceptor cell R8 | prepupal stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 69.0, 0.015],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.399868488120951, 'y': 65.3108075845233}},\n",
       " {'data': {'id': 'some die during P-stage',\n",
       "   'label': 'some die during P-stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.009, 350.0, 0.077],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.15287523373973, 'y': 18.053951026822745}},\n",
       " {'data': {'id': 'some die during pupal stage',\n",
       "   'label': 'some die during pupal stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.08, 435.0, 0.096],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.815287592859836, 'y': 7.72087050200615}},\n",
       " {'data': {'id': 'chemical sensitive',\n",
       "   'label': 'chemical sensitive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.059, 423.0, 0.093],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.704183453178542, 'y': 91.92356104620981}},\n",
       " {'data': {'id': 'abnormal flight, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'label': 'abnormal flight, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 320.0, 0.07],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 9.806874639508777, 'y': 0.00913355978970909}},\n",
       " {'data': {'id': 'partially lethal - majority die, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'label': 'partially lethal - majority die, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.047, 418.0, 0.092],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.52206223035694, 'y': 97.80991016904142}},\n",
       " {'data': {'id': 'some die during pupal stage, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'label': 'some die during pupal stage, with Scer\\\\GAL4[pnr-MD237]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.086, 438.0, 0.096],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.27046752236684, 'y': 0.30958849692858115}},\n",
       " {'data': {'id': 'abnormal neurophysiology | adult stage | conditional, with Scer\\\\GAL4[Or47b.7.467]',\n",
       "   'label': 'abnormal neurophysiology | adult stage | conditional, with Scer\\\\GAL4[Or47b.7.467]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 257.0, 0.056],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.763797453890963, 'y': 40.84535941257346}},\n",
       " {'data': {'id': 'adult olfactory receptor neuron Or47b | adult stage | conditional, with Scer\\\\GAL4[Or47b.7.467]',\n",
       "   'label': 'adult olfactory receptor neuron Or47b | adult stage | conditional, with Scer\\\\GAL4[Or47b.7.467]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 256.0, 0.056],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.711869233514037, 'y': 52.61505121916007}},\n",
       " {'data': {'id': 'abdominal sternite bristle, with Scer\\\\GAL4[-]:',\n",
       "   'label': 'abdominal sternite bristle, with Scer\\\\GAL4[-]:',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 43.0, 0.009],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.98437502556195, 'y': 56.2914079336005}},\n",
       " {'data': {'id': 'embryonic/larval fat body | wandering third instar larval stage, with Met[W3], Scer\\\\GAL4[Adh.PF]',\n",
       "   'label': 'embryonic/larval fat body | wandering third instar larval stage, with Met[W3], Scer\\\\GAL4[Adh.PF]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 62.0, 0.014],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.76513227736956, 'y': 44.904527034306696}},\n",
       " {'data': {'id': 'lethal - all die before end of larval stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'label': 'lethal - all die before end of larval stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.006, 326.0, 0.072],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.76991430866501, 'y': 38.44290508133238}},\n",
       " {'data': {'id': 'lethal | larval stage, with Scer\\\\GAL4[Act.PU]',\n",
       "   'label': 'lethal | larval stage, with Scer\\\\GAL4[Act.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 228.0, 0.05],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.70712773938445, 'y': 54.468662514935986}},\n",
       " {'data': {'id': 'male genitalia, with Scer\\\\GAL4[-]:',\n",
       "   'label': 'male genitalia, with Scer\\\\GAL4[-]:',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 13.0, 0.003],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.42664827306695, 'y': 99.79165437657413}},\n",
       " {'data': {'id': 'partially lethal - majority die | larval stage, with Scer\\\\GAL4[Act.PU]',\n",
       "   'label': 'partially lethal - majority die | larval stage, with Scer\\\\GAL4[Act.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 42.0, 0.009],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.249338610560612, 'y': 15.751794543635134}},\n",
       " {'data': {'id': 'partially lethal - majority die, with Scer\\\\GAL4[-]:',\n",
       "   'label': 'partially lethal - majority die, with Scer\\\\GAL4[-]:',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 259.0, 0.057],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.21685985434434, 'y': 24.051350251948634}},\n",
       " {'data': {'id': 'some die during larval stage, with Scer\\\\GAL4[-]:',\n",
       "   'label': 'some die during larval stage, with Scer\\\\GAL4[-]:',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 207.0, 0.045],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.63802290390953, 'y': 49.27819863952254}},\n",
       " {'data': {'id': 'some die during larval stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'label': 'some die during larval stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 302.0, 0.066],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.303537097229857, 'y': 11.057410235077347}},\n",
       " {'data': {'id': 'visible, with Scer\\\\GAL4[-]:',\n",
       "   'label': 'visible, with Scer\\\\GAL4[-]:',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 291.0, 0.064],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.388974317761907, 'y': 37.359013763921865}},\n",
       " {'data': {'id': 'embryonic/larval fat body | wandering third instar larval stage, with Met[UAS.cBa], Scer\\\\GAL4[Adh.PF]',\n",
       "   'label': 'embryonic/larval fat body | wandering third instar larval stage, with Met[UAS.cBa], Scer\\\\GAL4[Adh.PF]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 64.0, 0.014],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.231729376698116, 'y': 36.168322018266544}},\n",
       " {'data': {'id': 'FBgn0261703',\n",
       "   'label': 'FBgn0261703',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 24.14326849819525, 'y': 10.252922992552026}},\n",
       " {'data': {'id': 'GO0030016:myofibril, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'label': 'GO0030016:myofibril, with Scer\\\\GAL4[Mef2.PR]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.05, 419.0, 0.092],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.6739852205554, 'y': 83.37385809485919}},\n",
       " {'data': {'id': 'lethal | larval stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'label': 'lethal | larval stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 65.0, 0.014],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.790545739919065, 'y': 58.11020012556949}},\n",
       " {'data': {'id': 'lethal | pharate adult stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'label': 'lethal | pharate adult stage, with Scer\\\\GAL4[Tub.PU]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 66.0, 0.015],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.67575941219387, 'y': 25.75293515443485}},\n",
       " {'data': {'id': 'short lived, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'label': 'short lived, with Scer\\\\GAL4[Act5C.PI]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 303.0, 0.067],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.80995225547784, 'y': 65.36374508283339}},\n",
       " {'data': {'id': 'FBgn0023076',\n",
       "   'label': 'FBgn0023076',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'dmelanogaster'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 57.33024776799818, 'y': 65.56956307657211}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Clk[out]',\n",
       "   'label': 'abnormal locomotor rhythm, with Clk[out]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 67.0, 0.015],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.000952988098693, 'y': 33.42277571864375}},\n",
       " {'data': {'id': 'abnormal temperature response, with Clk[out]',\n",
       "   'label': 'abnormal temperature response, with Clk[out]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 68.0, 0.015],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.41984296082457, 'y': 23.767500929330886}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[to.PD]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[to.PD]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 192.0, 0.042],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 2.1287670866827435, 'y': 15.223294267911946}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[yolk]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[yolk]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 75.0, 0.016],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.035643753817737, 'y': 81.77465412490463}},\n",
       " {'data': {'id': 'abnormal feeding behavior, with Scer\\\\GAL4[Orco.2.642.Hsim\\\\VP22]',\n",
       "   'label': 'abnormal feeding behavior, with Scer\\\\GAL4[Orco.2.642.Hsim\\\\VP22]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 50.0, 0.011],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.210838253919082, 'y': 61.75732616463711}},\n",
       " {'data': {'id': 'abnormal feeding behavior, with Scer\\\\GAL4[to.PD]',\n",
       "   'label': 'abnormal feeding behavior, with Scer\\\\GAL4[to.PD]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 186.0, 0.041],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.952199606361999, 'y': 65.12542816996334}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[to.PD]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[to.PD]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 44.0, 0.01],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.958546052804717, 'y': 96.69509244262049}},\n",
       " {'data': {'id': 'abnormal circadian behavior, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'label': 'abnormal circadian behavior, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 287.0, 0.063],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 6.479612874545737, 'y': 7.280932922597339}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[Or22a.8197]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[Or22a.8197]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 196.0, 0.043],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.7556836511727982, 'y': 90.54774142902521}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[Orco.2.642.Hsim\\\\VP22]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[Orco.2.642.Hsim\\\\VP22]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 212.0, 0.047],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.77821697373156, 'y': 19.902182103593592}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[tim.PE]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[tim.PE]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 288.0, 0.063],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.84968365664831, 'y': 66.15003568571588}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[P0.5.Pdf]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[P0.5.Pdf]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 260.0, 0.057],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.975544757057044, 'y': 15.552773241373352}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.009, 352.0, 0.077],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.262911274448054, 'y': 69.57826452406147}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[tim.PE]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[tim.PE]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.013, 364.0, 0.08],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.36676970858778, 'y': 93.93526637039896}},\n",
       " {'data': {'id': 'abnormal sleep, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'label': 'abnormal sleep, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 255.0, 0.056],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 82.26012395642341, 'y': 93.50710389270505}},\n",
       " {'data': {'id': 'Pdf neuron',\n",
       "   'label': 'Pdf neuron',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 235.0, 0.052],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.75200785506324, 'y': 22.792807127368754}},\n",
       " {'data': {'id': 'abnormal circadian rhythm | adult stage, with Clk[out]',\n",
       "   'label': 'abnormal circadian rhythm | adult stage, with Clk[out]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 182.0, 0.04],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 87.50681940754065, 'y': 94.24948018336569}},\n",
       " {'data': {'id': 'abnormal circadian rhythm',\n",
       "   'label': 'abnormal circadian rhythm',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.007, 335.0, 0.074],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.05348216684672, 'y': 42.23661409549548}},\n",
       " {'data': {'id': 'abnormal eclosion rhythm',\n",
       "   'label': 'abnormal eclosion rhythm',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 308.0, 0.068],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 94.02827198718138, 'y': 5.607137950175234}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Clk[Z3-0084])',\n",
       "   'label': 'abnormal locomotor rhythm (with Clk[Z3-0084])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 168.0, 0.037],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.02391538026917, 'y': 48.37154301529606}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Clk[ar])',\n",
       "   'label': 'abnormal locomotor rhythm (with Clk[ar])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 189.0, 0.042],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 17.93402712468504, 'y': 97.65906193414827}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm | dominant',\n",
       "   'label': 'abnormal locomotor rhythm | dominant',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 277.0, 0.061],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.5320638846737578, 'y': 88.60710982889375}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm | semidominant',\n",
       "   'label': 'abnormal locomotor rhythm | semidominant',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 263.0, 0.058],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.40752686454672, 'y': 98.22737920136197}},\n",
       " {'data': {'id': 'abnormal sleep',\n",
       "   'label': 'abnormal sleep',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.018, 381.0, 0.084],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.20528192136576, 'y': 12.285900001505723}},\n",
       " {'data': {'id': 'abnormal visual behavior',\n",
       "   'label': 'abnormal visual behavior',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.023, 387.0, 0.085],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.70039048572994, 'y': 17.155715270304118}},\n",
       " {'data': {'id': 'head',\n",
       "   'label': 'head',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.021, 385.0, 0.085],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.70020655435576, 'y': 63.57230241760775}},\n",
       " {'data': {'id': 'increased cell death | progressive',\n",
       "   'label': 'increased cell death | progressive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 159.0, 0.035],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 16.794039605113078, 'y': 69.5008234919395}},\n",
       " {'data': {'id': 'larval LN period neuron',\n",
       "   'label': 'larval LN period neuron',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 247.0, 0.054],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 98.75924888728608, 'y': 24.014275560756314}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[out])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[out])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 165.0, 0.036],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.4475927360684, 'y': 95.52395811965854}},\n",
       " {'data': {'id': 'pupal Pdf neuron',\n",
       "   'label': 'pupal Pdf neuron',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 166.0, 0.036],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.10292215621698, 'y': 69.50047415500855}},\n",
       " {'data': {'id': 'abnormal temperature response',\n",
       "   'label': 'abnormal temperature response',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.012, 361.0, 0.079],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.29119286437214, 'y': 85.57389441058312}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[cry.PE]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[cry.PE]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 219.0, 0.048],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 85.28201994390476, 'y': 96.03345819575168}},\n",
       " {'data': {'id': 'abnormal circadian rhythm, with Scer\\\\GAL4[cry.PZ]',\n",
       "   'label': 'abnormal circadian rhythm, with Scer\\\\GAL4[cry.PZ]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 227.0, 0.05],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 91.65899759862707, 'y': 16.305111215826617}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm | adult stage, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'label': 'abnormal locomotor rhythm | adult stage, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 299.0, 0.066],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.5517120378470084, 'y': 92.03916584936906}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[cry.PE]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[cry.PE]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.002, 262.0, 0.058],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.662282139926017, 'y': 56.393750242485694}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[cry.PZ]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[cry.PZ]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 298.0, 0.065],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.29181746382318, 'y': 16.77680341219817}},\n",
       " {'data': {'id': 'lethal, with Scer\\\\GAL4[unspecified]',\n",
       "   'label': 'lethal, with Scer\\\\GAL4[unspecified]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.004, 306.0, 0.067],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.84718261536486, 'y': 49.705578853463685}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[MJ162A]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[MJ162A]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 45.0, 0.01],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.0547319156842, 'y': 9.52584756217021}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[P2.4.Pdf]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 231.0, 0.051],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 14.264434663247162, 'y': 73.48440544080503}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[cry.PE]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[cry.PE]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.001, 210.0, 0.046],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.92816461747861, 'y': 40.90487061568252}},\n",
       " {'data': {'id': 'viable, with Scer\\\\GAL4[cry.PZ]',\n",
       "   'label': 'viable, with Scer\\\\GAL4[cry.PZ]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 46.0, 0.01],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.39430607828817, 'y': 52.35196056083602}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Clk[Jrk])',\n",
       "   'label': 'abnormal locomotor rhythm (with Clk[Jrk])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 47.0, 0.01],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.459029374222827, 'y': 74.54181324322968}},\n",
       " {'data': {'id': 'abnormal circadian rhythm | adult stage',\n",
       "   'label': 'abnormal circadian rhythm | adult stage',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.003, 294.0, 0.065],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 5.757013671424942, 'y': 25.012937169491966}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm (with Df(3L)HnD-1)',\n",
       "   'label': 'abnormal locomotor rhythm (with Df(3L)HnD-1)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 48.0, 0.011],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.898550200163431, 'y': 28.958380932803273}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm | recessive',\n",
       "   'label': 'abnormal locomotor rhythm | recessive',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.005, 316.0, 0.069],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.426372738171952, 'y': 59.393684325522244}},\n",
       " {'data': {'id': 'adult heart',\n",
       "   'label': 'adult heart',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.014, 368.0, 0.081],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.222484737160764, 'y': 11.1150267537515}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Scer\\\\GAL4[tim.UAS]',\n",
       "   'label': 'abnormal locomotor rhythm, with Scer\\\\GAL4[tim.UAS]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.008, 349.0, 0.077],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 14.241047065512005, 'y': 11.913490336734112}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S258A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S258A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 49.0, 0.011],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 84.61071598832137, 'y': 15.820302867901958}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S476A.S482A.S487A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S476A.S482A.S487A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 51.0, 0.011],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 8.787398641614528, 'y': 43.283062907319824}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S476A.S482A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S476A.S482A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 58.0, 0.013],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.866769491462467, 'y': 46.55425895229084}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S487A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S487A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 52.0, 0.011],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.15017138497382, 'y': 63.48326093272897}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S525A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S525A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 53.0, 0.012],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.89230609203318, 'y': 16.41036574805487}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S859A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S859A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 54.0, 0.012],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.8821974812128364, 'y': 87.63965301231579}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Clk[S912A.Tag:V5])',\n",
       "   'label': 'abnormal circadian rhythm (with Clk[S912A.Tag:V5])',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 55.0, 0.012],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 11.353200568513524, 'y': 30.110430800346254}},\n",
       " {'data': {'id': 'abnormal circadian rhythm (with Df(3L)BSC631)',\n",
       "   'label': 'abnormal circadian rhythm (with Df(3L)BSC631)',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 56.0, 0.012],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 7.4658424869932505, 'y': 50.61455701613926}},\n",
       " {'data': {'id': 'abnormal circadian rhythm | adult stage, with Clk[Tag:V5.WT]',\n",
       "   'label': 'abnormal circadian rhythm | adult stage, with Clk[Tag:V5.WT]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 57.0, 0.013],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.63462465497322, 'y': 3.183160928954498}},\n",
       " {'data': {'id': 'abnormal circadian rhythm | adult stage, with Clk[Δ657-707.Tag:V5]',\n",
       "   'label': 'abnormal circadian rhythm | adult stage, with Clk[Δ657-707.Tag:V5]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 130.0, 0.029],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.76043172928827, 'y': 90.50651321965168}},\n",
       " {'data': {'id': 'abnormal locomotor rhythm, with Clk[15A.Tag:V5]',\n",
       "   'label': 'abnormal locomotor rhythm, with Clk[15A.Tag:V5]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 152.0, 0.033],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 92.91532770933524, 'y': 75.13823547301018}},\n",
       " {'data': {'id': 'abnormal temperature response, with Clk[15A.Tag:V5]',\n",
       "   'label': 'abnormal temperature response, with Clk[15A.Tag:V5]',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': [None, None, 0.0, 153.0, 0.034],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.57175281861632, 'y': 53.01516309261564}},\n",
       " {'data': {'id': 'Ahrr',\n",
       "   'label': 'Ahrr',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 36.78570261121064, 'y': 24.177481801354837}},\n",
       " {'data': {'id': 'MP:0005377',\n",
       "   'label': 'MP:0005377',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['hearing/vestibular/ear phenotype',\n",
       "    ['MP:0000001'],\n",
       "    0.044,\n",
       "    9.0,\n",
       "    0.069],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.36434072879268, 'y': 57.48792572426949}},\n",
       " {'data': {'id': 'MP:0004738',\n",
       "   'label': 'MP:0004738',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['abnormal auditory brainstem response',\n",
       "    ['MP:0006335'],\n",
       "    0.041,\n",
       "    8.0,\n",
       "    0.062],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 96.14625533276565, 'y': 97.63737734003223}},\n",
       " {'data': {'id': 'MP:0002757',\n",
       "   'label': 'MP:0002757',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['decreased vertical activity', [], 0.024, 5.0, 0.038],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 99.44746109518394, 'y': 73.01349351171403}},\n",
       " {'data': {'id': 'MP:0005561',\n",
       "   'label': 'MP:0005561',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['increased mean corpuscular hemoglobin',\n",
       "    ['MP:0001589'],\n",
       "    0.032,\n",
       "    7.0,\n",
       "    0.054],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 95.8438637263692, 'y': 1.6516037603693468}},\n",
       " {'data': {'id': 'MP:0005564',\n",
       "   'label': 'MP:0005564',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['increased hemoglobin content',\n",
       "    ['MP:0005563'],\n",
       "    0.018,\n",
       "    3.0,\n",
       "    0.023],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 3.0235609617826187, 'y': 53.40039694962271}},\n",
       " {'data': {'id': 'MP:0000186',\n",
       "   'label': 'MP:0000186',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['decreased circulating HDL cholesterol level',\n",
       "    [],\n",
       "    0.02,\n",
       "    4.0,\n",
       "    0.031],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 15.036439125995166, 'y': 82.33314290865734}},\n",
       " {'data': {'id': 'MP:0001325',\n",
       "   'label': 'MP:0001325',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['abnormal retina morphology',\n",
       "    ['MP:0002864'],\n",
       "    0.03,\n",
       "    6.0,\n",
       "    0.046],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.84194329933501, 'y': 23.873141562996693}},\n",
       " {'data': {'id': 'MP:0011396',\n",
       "   'label': 'MP:0011396',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['abnormal sleep behavior', ['MP:0004924'], 0.013, 2.0, 0.015],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 83.50649822049783, 'y': 40.336162297769874}},\n",
       " {'data': {'id': 'Hsp90ab1',\n",
       "   'label': 'Hsp90ab1',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 31.090410896998527, 'y': 62.36589565669377}},\n",
       " {'data': {'id': 'Ahr',\n",
       "   'label': 'Ahr',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 56.42528684942618, 'y': 71.24945653665966}},\n",
       " {'data': {'id': 'Aip',\n",
       "   'label': 'Aip',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 55.90568992616199, 'y': 0.16431356995224577}},\n",
       " {'data': {'id': 'Ptges3',\n",
       "   'label': 'Ptges3',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 21.685156366249316, 'y': 86.4307485294471}},\n",
       " {'data': {'id': 'Arnt',\n",
       "   'label': 'Arnt',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 35.46943399524945, 'y': 77.13090830514344}},\n",
       " {'data': {'id': 'Arnt2',\n",
       "   'label': 'Arnt2',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'mouse'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 59.857980491273025, 'y': 44.903795279488115}},\n",
       " {'data': {'id': 'ahrra',\n",
       "   'label': 'ahrra',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 62.92212060918392, 'y': 78.69043747362649}},\n",
       " {'data': {'id': 'ahrrb',\n",
       "   'label': 'ahrrb',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 38.66458327054353, 'y': 68.2571835004774}},\n",
       " {'data': {'id': 'hsp90ab1',\n",
       "   'label': 'hsp90ab1',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 36.486744966130274, 'y': 12.7821787008351}},\n",
       " {'data': {'id': 'ahr1a',\n",
       "   'label': 'ahr1a',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 22.223919710146337, 'y': 88.72740117109728}},\n",
       " {'data': {'id': 'aip',\n",
       "   'label': 'aip',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 67.55401213469985, 'y': 31.014999411751866}},\n",
       " {'data': {'id': 'ptges3a',\n",
       "   'label': 'ptges3a',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 29.045329528041528, 'y': 18.8741666415015}},\n",
       " {'data': {'id': 'ptges3b',\n",
       "   'label': 'ptges3b',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 75.85543600743961, 'y': 26.794311760613976}},\n",
       " {'data': {'id': 'WBGene00000096',\n",
       "   'label': 'WBGene00000096',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'celegans'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 71.7318283387965, 'y': 92.25275173858695}},\n",
       " {'data': {'id': 'WBPhenotype:0000625',\n",
       "   'label': 'WBPhenotype:0000625',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['synaptogenesis variant',\n",
       "    ['WBPhenotype:0000623'],\n",
       "    0.004,\n",
       "    7.0,\n",
       "    0.012],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 89.21817465778018, 'y': 26.74554747122616}},\n",
       " {'data': {'id': 'WBPhenotype:0001331',\n",
       "   'label': 'WBPhenotype:0001331',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['motor neuron morphology variant',\n",
       "    ['WBPhenotype:0000905'],\n",
       "    0.0,\n",
       "    2.0,\n",
       "    0.004],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 12.10754904376147, 'y': 49.98357363883348}},\n",
       " {'data': {'id': 'WBPhenotype:0001510',\n",
       "   'label': 'WBPhenotype:0001510',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['neuronal cell fate specification variant',\n",
       "    ['WBPhenotype:0000216'],\n",
       "    0.015,\n",
       "    15.0,\n",
       "    0.026],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 80.18244905165756, 'y': 2.6258949535642784}},\n",
       " {'data': {'id': 'WBPhenotype:0002261',\n",
       "   'label': 'WBPhenotype:0002261',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['neurite connectivity defective',\n",
       "    ['WBPhenotype:0000623'],\n",
       "    0.0,\n",
       "    1.0,\n",
       "    0.002],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.587323999269975, 'y': 67.23990863253073}},\n",
       " {'data': {'id': 'WBGene00000915',\n",
       "   'label': 'WBGene00000915',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'celegans'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 77.24428992338879, 'y': 33.58207640368695}},\n",
       " {'data': {'id': 'WBPhenotype:0000012',\n",
       "   'label': 'WBPhenotype:0000012',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['dauer constitutive',\n",
       "    ['WBPhenotype:0000637'],\n",
       "    0.002,\n",
       "    5.0,\n",
       "    0.009],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 1.7147249389789931, 'y': 96.3891872058471}},\n",
       " {'data': {'id': 'WBPhenotype:0000127',\n",
       "   'label': 'WBPhenotype:0000127',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['dauer recovery variant',\n",
       "    ['WBPhenotype:0000159'],\n",
       "    0.007,\n",
       "    12.0,\n",
       "    0.021],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 86.26585127199561, 'y': 84.38871605531226}},\n",
       " {'data': {'id': 'WBPhenotype:0000646',\n",
       "   'label': 'WBPhenotype:0000646',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['sluggish', ['WBPhenotype:0001213'], 0.005, 9.0, 0.016],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 10.138330485136702, 'y': 15.087434756841722}},\n",
       " {'data': {'id': 'WBPhenotype:0001048',\n",
       "   'label': 'WBPhenotype:0001048',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['odorant chemosensory response variant',\n",
       "    ['WBPhenotype:0001040'],\n",
       "    0.005,\n",
       "    8.0,\n",
       "    0.014],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 90.13509990545042, 'y': 84.19515059358062}},\n",
       " {'data': {'id': 'WBPhenotype:0001434',\n",
       "   'label': 'WBPhenotype:0001434',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['chemotaxis variant',\n",
       "    ['WBPhenotype:0001040'],\n",
       "    0.0,\n",
       "    3.0,\n",
       "    0.005],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.24656249958711, 'y': 83.02470325717258}},\n",
       " {'data': {'id': 'WBPhenotype:0000644',\n",
       "   'label': 'WBPhenotype:0000644',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['paralyzed', ['WBPhenotype:0001213'], 0.003, 6.0, 0.011],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 18.834396390024956, 'y': 57.851202345900134}},\n",
       " {'data': {'id': 'WBPhenotype:0001012',\n",
       "   'label': 'WBPhenotype:0001012',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['organism pathogen response variant',\n",
       "    ['WBPhenotype:0000738'],\n",
       "    0.023,\n",
       "    16.0,\n",
       "    0.028],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 81.33324867532181, 'y': 80.95246539911349}},\n",
       " {'data': {'id': 'WBPhenotype:0000055',\n",
       "   'label': 'WBPhenotype:0000055',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['early larval arrest',\n",
       "    ['WBPhenotype:0000059'],\n",
       "    0.006,\n",
       "    11.0,\n",
       "    0.019],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 13.979072924089543, 'y': 3.3410446341763933}},\n",
       " {'data': {'id': 'WBPhenotype:0000668',\n",
       "   'label': 'WBPhenotype:0000668',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['endomitotic oocytes',\n",
       "    ['WBPhenotype:0001260'],\n",
       "    0.025,\n",
       "    17.0,\n",
       "    0.03],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.647232272121462, 'y': 66.19084980316254}},\n",
       " {'data': {'id': 'WBPhenotype:0001943',\n",
       "   'label': 'WBPhenotype:0001943',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['rachis wide', ['WBPhenotype:0001940'], 0.034, 20.0, 0.035],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.36313515719044, 'y': 15.788077380311949}},\n",
       " {'data': {'id': 'WBPhenotype:0001947',\n",
       "   'label': 'WBPhenotype:0001947',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['diplotene progression during oogenesis variant',\n",
       "    ['WBPhenotype:0001824'],\n",
       "    0.026,\n",
       "    18.0,\n",
       "    0.032],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 4.522930527988118, 'y': 6.146619048683732}},\n",
       " {'data': {'id': 'WBPhenotype:0001349',\n",
       "   'label': 'WBPhenotype:0001349',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['protein phosphorylation variant',\n",
       "    ['WBPhenotype:0001382'],\n",
       "    0.005,\n",
       "    10.0,\n",
       "    0.018],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 0.10688790667927606, 'y': 6.836462589352854}},\n",
       " {'data': {'id': 'WBPhenotype:0001925',\n",
       "   'label': 'WBPhenotype:0001925',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['oocytes disorganized',\n",
       "    ['WBPhenotype:0001260'],\n",
       "    0.035,\n",
       "    21.0,\n",
       "    0.037],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.15541648297355, 'y': 43.26675053237994}},\n",
       " {'data': {'id': 'WBGene00016966',\n",
       "   'label': 'WBGene00016966',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'celegans'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 61.20175012183974, 'y': 82.64805618754896}},\n",
       " {'data': {'id': 'nan',\n",
       "   'label': 'nan',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': None,\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 19.039396605533444, 'y': 24.562425996449345}},\n",
       " {'data': {'id': 'WBGene00022599',\n",
       "   'label': 'WBGene00022599',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'celegans'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 61.39955344648559, 'y': 39.53958222146498}},\n",
       " {'data': {'id': 'WBPhenotype:0000295',\n",
       "   'label': 'WBPhenotype:0000295',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['thermotolerance increased',\n",
       "    ['WBPhenotype:0000146'],\n",
       "    0.011,\n",
       "    14.0,\n",
       "    0.025],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 88.2795990513529, 'y': 75.12592764072154}},\n",
       " {'data': {'id': 'WBPhenotype:0001789',\n",
       "   'label': 'WBPhenotype:0001789',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['temperature influenced life span variant',\n",
       "    ['WBPhenotype:0000039', 'WBPhenotype:0000146'],\n",
       "    0.002,\n",
       "    4.0,\n",
       "    0.007],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 97.93211099553442, 'y': 15.629768630792483}},\n",
       " {'data': {'id': 'WBPhenotype:0001403',\n",
       "   'label': 'WBPhenotype:0001403',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['antibody subcellular localization variant',\n",
       "    ['WBPhenotype:0000424'],\n",
       "    0.032,\n",
       "    19.0,\n",
       "    0.033],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.34241905279268, 'y': 76.63880076304304}},\n",
       " {'data': {'id': 'WBGene00000095',\n",
       "   'label': 'WBGene00000095',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'celegans'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 59.16782324839741, 'y': 12.823417117507807}},\n",
       " {'data': {'id': 'WBPhenotype:0001989',\n",
       "   'label': 'WBPhenotype:0001989',\n",
       "   'indent': 'phenotype',\n",
       "   'metadata': ['hypoxia hypersensitive',\n",
       "    ['WBPhenotype:0002423'],\n",
       "    0.007,\n",
       "    13.0,\n",
       "    0.023],\n",
       "   'size': 1,\n",
       "   'fontsize': '0.5px'},\n",
       "  'classes': 'purple',\n",
       "  'selectable': True,\n",
       "  'grabbable': False,\n",
       "  'position': {'x': 93.34266956802091, 'y': 36.787163494252184}}]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "8b011bc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'data': {'id': 'arnt',\n",
       "   'label': 'arnt',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 41.83495621677152, 'y': 48.064603068407585}},\n",
       " {'data': {'id': 'arnt2',\n",
       "   'label': 'arnt2',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 39.55501146034899, 'y': 64.72956443328935}},\n",
       " {'data': {'id': 'arnt2',\n",
       "   'label': 'arnt2',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 39.55501146034899, 'y': 64.72956443328935}},\n",
       " {'data': {'id': 'ahrra',\n",
       "   'label': 'ahrra',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 62.92212060918392, 'y': 78.69043747362649}},\n",
       " {'data': {'id': 'ahrrb',\n",
       "   'label': 'ahrrb',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 38.66458327054353, 'y': 68.2571835004774}},\n",
       " {'data': {'id': 'hsp90ab1',\n",
       "   'label': 'hsp90ab1',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 36.486744966130274, 'y': 12.7821787008351}},\n",
       " {'data': {'id': 'ahr1a',\n",
       "   'label': 'ahr1a',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 22.223919710146337, 'y': 88.72740117109728}},\n",
       " {'data': {'id': 'aip',\n",
       "   'label': 'aip',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 67.55401213469985, 'y': 31.014999411751866}},\n",
       " {'data': {'id': 'ptges3a',\n",
       "   'label': 'ptges3a',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 29.045329528041528, 'y': 18.8741666415015}},\n",
       " {'data': {'id': 'ptges3b',\n",
       "   'label': 'ptges3b',\n",
       "   'indent': 'ortholog',\n",
       "   'size': 2,\n",
       "   'fontsize': '1px',\n",
       "   'organism': 'zebrafish'},\n",
       "  'classes': 'ortholog',\n",
       "  'selectable': True,\n",
       "  'position': {'x': 75.85543600743961, 'y': 26.794311760613976}}]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "new_test = []\n",
    "\n",
    "for i in new_element1:\n",
    "    if i.get('data').get('organism') == 'zebrafish':\n",
    "        new_test.append(i)\n",
    "new_test"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4965f0c8",
   "metadata": {},
   "source": [
    "## Define organism and colors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1a173c6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#organisms list:\n",
    "organism = ['dmelanogaster','mouse','celegans','zebrafish']\n",
    "# https://plotly.com/python/discrete-color/\n",
    "col_swatch = [px.colors.qualitative.Safe[8], px.colors.qualitative.Set1[2],\n",
    "              px.colors.qualitative.Set1[7],px.colors.qualitative.Set1[4]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c10fed1",
   "metadata": {},
   "source": [
    "## Function to caluculate the number of phenotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "efd8b052",
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate the right phenotype number\n",
    "def calculate_no_phenotypes (df):\n",
    "    no_celegans = (df.groupby(['Organism']).count())['Enriched_Phenotypes'][0]\n",
    "    no_dmelanogaster = (df.groupby(['Organism']).count())['Enriched_Phenotypes'][1]\n",
    "    no_mouse = (df.groupby(['Organism']).count())['Enriched_Phenotypes'][2]\n",
    "    no_zebrafish = (df.groupby(['Organism']).count())['Enriched_Phenotypes'][3]\n",
    "    totalPhenotype = [no_celegans,no_dmelanogaster,no_mouse,no_zebrafish]\n",
    "    org_totalPhenotype = dict(zip(organism,totalPhenotype))\n",
    "    return org_totalPhenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "1b4451e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dmelanogaster': 217, 'mouse': 350, 'celegans': 57, 'zebrafish': 0}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_totalPhenotype = calculate_no_phenotypes(upper_df1)\n",
    "org_totalPhenotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "eeb1640d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dmelanogaster': 330, 'mouse': 8634, 'celegans': 48, 'zebrafish': 9}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_totalPhenotype_1 = calculate_no_phenotypes(gene_df_1)\n",
    "org_totalPhenotype_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "39755145",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dmelanogaster': 865, 'mouse': 13523, 'celegans': 0, 'zebrafish': 0}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_totalPhenotype_2 = calculate_no_phenotypes(gene_df_2)\n",
    "org_totalPhenotype_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9aa800cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dmelanogaster': 5136, 'mouse': 57023, 'celegans': 2, 'zebrafish': 0}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_totalPhenotype_3 = calculate_no_phenotypes(gene_df_3)\n",
    "org_totalPhenotype_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7f7120ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dmelanogaster': 865, 'mouse': 13523, 'celegans': 0, 'zebrafish': 0}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_totalPhenotype_4 = calculate_no_phenotypes(gene_df_4)\n",
    "org_totalPhenotype_4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06a080cf",
   "metadata": {},
   "source": [
    "## Define app:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "074e5baf",
   "metadata": {},
   "outputs": [],
   "source": [
    "graph_stylesheet = [\n",
    "    # {\n",
    "    #     'selector': organism[i],       # colours based on organism\n",
    "    #     'style': {\"background-color\": col_swatch[i],\n",
    "    #               \"line-color\": col_swatch[i]\n",
    "    #               },\n",
    "    # }\n",
    "    # for i in range(len(organism))\n",
    "\n",
    "# colours based on organism\n",
    "    {\n",
    "        'selector': '[organism *= \"dmelanogaster\"]',\n",
    "        'style': {\n",
    "            'background-color': col_swatch[0],\n",
    "            'line-colour': col_swatch[0],\n",
    "        }\n",
    "    },\n",
    "    {\n",
    "        'selector': '[organism *= \"mouse\"]',\n",
    "        'style': {\n",
    "            'background-color': col_swatch[1],\n",
    "            'line-colour': col_swatch[1],\n",
    "        }\n",
    "    },\n",
    "    {\n",
    "        'selector': '[organism *= \"celegans\"]',\n",
    "        'style': {\n",
    "            'background-color': col_swatch[2],\n",
    "            'line-colour': col_swatch[2],\n",
    "        }\n",
    "    },\n",
    "    {\n",
    "        'selector': '[organism *= \"zebrafish\"]',\n",
    "        'style': {\n",
    "            'background-color': col_swatch[3],\n",
    "            'line-colour': col_swatch[3],\n",
    "        }\n",
    "    }\n",
    "]\n",
    "\n",
    "graph_stylesheet += [\n",
    "    {\n",
    "        'selector': 'nodes',\n",
    "        'style': {\n",
    "            'content': 'data(label)',\n",
    "            'width': 'data(size)',\n",
    "            'height': 'data(size)',\n",
    "            \"curve-style\": \"bezier\",\n",
    "            'font-size': 'data(fontsize)',\n",
    "        }\n",
    "    },\n",
    "    {\n",
    "        'selector': 'edges',\n",
    "        'style': {\n",
    "            'width': 'data(width)',\n",
    "            'line-color': 'data(color)',\n",
    "        }\n",
    "    },\n",
    "    {\n",
    "        'selector': '.purple',\n",
    "        'style': {\n",
    "            'background-color': '#C918F3',\n",
    "            'line-color': '#C918F3'\n",
    "        }\n",
    "    },\n",
    "\n",
    "    {\n",
    "        'selector': '.blue',\n",
    "        'style': {'background-color': '#18B6F3',\n",
    "                  'line-color': '#18B6F3'\n",
    "                  }\n",
    "    }\n",
    "\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "308b10c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define app\n",
    "app = dash.Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])\n",
    "# server = app.server"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "9da88d9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "navbar = dbc.NavbarSimple(\n",
    "    children=[\n",
    "        dbc.NavItem(\n",
    "            dbc.NavLink(\n",
    "                \"Article\",\n",
    "                href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5829367/\",\n",
    "            )\n",
    "        ),\n",
    "        dbc.NavItem(\n",
    "            dbc.NavLink(\n",
    "                \"Source Code\",\n",
    "                href=\"https://github.com/ivanajanickova/DARTpaths-vis\",\n",
    "            )\n",
    "        ),\n",
    "    ],\n",
    "    brand=\"Dart - Translation of Genotype to Phenotype by a Hierarchy of Model Organism - simple demo\", #title\n",
    "    brand_href=\"#\",\n",
    "    color=\"dark\",\n",
    "    dark=True,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "69db4a25",
   "metadata": {},
   "outputs": [],
   "source": [
    "topics_html = list()\n",
    "for topic_html in [\n",
    "    html.Span([str(i+1) + \": \" + organism[i]], style={\"color\": col_swatch[i]})\n",
    "    for i in range(len(organism))\n",
    "]:\n",
    "    topics_html.append(topic_html)\n",
    "    topics_html.append(html.Br())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f19e1465",
   "metadata": {},
   "source": [
    "## Body layout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "9031f644",
   "metadata": {},
   "outputs": [],
   "source": [
    "body_layout = dbc.Container(\n",
    "    [\n",
    "        dbc.Row(\n",
    "            [\n",
    "                dbc.Col(\n",
    "                    [\n",
    "                        dcc.Markdown(\n",
    "                            f\"\"\"\n",
    "                -----\n",
    "                ##### Description:\n",
    "                -----\n",
    "                For this demonstration, phenotypes from 4 model organism were used to visualize\n",
    "                the relationship between phenotypes from different organism related to a particular human\n",
    "                pathway. The pathways are preference from [REACTOME](https://reactome.org/PathwayBrowser/)\n",
    "                Phenotype from each organism is shown in different color on the map, as shown on the right.\n",
    "                \"\"\"\n",
    "                        )\n",
    "                    ],\n",
    "                    sm=12,\n",
    "                    md=8,\n",
    "                ),\n",
    "                dbc.Col(\n",
    "                    [\n",
    "                        dcc.Markdown(\n",
    "                            \"\"\"\n",
    "                -----\n",
    "                ##### Model oganisms:\n",
    "                -----\n",
    "                \"\"\"\n",
    "                        ),\n",
    "                        html.Div(\n",
    "                            topics_html,\n",
    "                            style={\n",
    "                                \"fontSize\": 11,\n",
    "                                \"height\": \"100px\",\n",
    "                                \"overflow\": \"auto\",\n",
    "                            },\n",
    "                        ),\n",
    "                    ],\n",
    "                    sm=12,\n",
    "                    md=4,\n",
    "                )\n",
    "            ]\n",
    "        ),\n",
    "        dbc.Row(\n",
    "            [\n",
    "                dcc.Markdown(\n",
    "                    \"\"\"\n",
    "            -----\n",
    "            ##### Node representation\n",
    "            There are three kinds of node in the graph. The biggest nodes in the center indicate human genes in a chosen \n",
    "            pathway. These nodes are linked directly with the second type of node which indicate orthologs from a model \n",
    "            organism. Node color indicates which organism the orthologs belong to. \n",
    "            \n",
    "            The third type of node in two sides is the phenotype enrichment of organism model\n",
    "            \n",
    "            The edges show the relationship of 1. phenotypes and its related ortholog gene in organism 2. ortholog genes in \n",
    "            organism and its human gene in the same molecular pathway in Reactome.\n",
    "\n",
    "            Use the filters to choose the phenotype with:\n",
    "            * certain numbers of pathway/ pathway level \n",
    "            * certain numbers of model organism you want to investigate\n",
    "            \n",
    "            The slider is used to explore 3 level of visualisation: human gene only, human gene and orthologs or the whole graph. Toggle to explore different visualisation options.\n",
    "            \n",
    "            -----\n",
    "            \"\"\"\n",
    "                )\n",
    "            ]\n",
    "        ),\n",
    "        dbc.Row(\n",
    "            [\n",
    "                dbc.Col(\n",
    "                    [\n",
    "                        dbc.Row(\n",
    "                            [\n",
    "                                cyto.Cytoscape(\n",
    "                                    id='cytoscape-phenotype',\n",
    "                                    layout={\"name\": \"preset\"},\n",
    "                                    style={\"width\": \"100%\", \"height\": \"800px\"},\n",
    "                                    elements= element,   #change\n",
    "                                    stylesheet=graph_stylesheet,\n",
    "                                    #style={'width': '100%', 'height': '95vh'}\n",
    "                                    minZoom = 0.6\n",
    "                                )\n",
    "                            ]\n",
    "                        ),\n",
    "                        dbc.Row(\n",
    "                            [\n",
    "                                dbc.Alert(\n",
    "                                    id='phenotype-data',\n",
    "                                    children='Click on a node to see its details here',\n",
    "                                    color='secondary'\n",
    "                                )\n",
    "                            ]\n",
    "                        )\n",
    "                    ],\n",
    "                    sm=12,\n",
    "                    md=8\n",
    "                ),\n",
    "                dbc.Col(\n",
    "                    [\n",
    "                        dbc.Label('Search phenotype for a gene: ', html_for='gene_search'),\n",
    "                        dbc.Input(type='search', id='gene_search', placeholder=\"Enter a gene ID here\"),\n",
    "                        dbc.FormText('The gene is not available in the pathway', color='secondary'),\n",
    "                        html.Div(style={'padding': 20}), #Space\n",
    "\n",
    "                        dbc.Row(\n",
    "                            [\n",
    "                                dbc.Col(\n",
    "                                    [\n",
    "                                        html.H4(dbc.Badge(\n",
    "                                            \"Level of pathway(s):\",\n",
    "                                            color=\"success\",\n",
    "                                            className=\"ms-1\",\n",
    "\n",
    "                                        )),\n",
    "\n",
    "                                        dcc.Dropdown(\n",
    "                                            id=\"pathway_level\",\n",
    "                                            #options=[{\"label\": k, \"value\": k} for k in range(1,3)],\n",
    "                                            options=[{\"label\": 'lower level', \"value\": 1},{\"label\": 'upper level', \"value\": 2}],\n",
    "                                            clearable=False,\n",
    "                                            value= 2,\n",
    "                                            style={\"width\": \"150px\"},\n",
    "                                    )\n",
    "                                    ]),\n",
    "                                dbc.Col(\n",
    "                                    [\n",
    "                                        #Irene\n",
    "                                        dbc.Badge(           # dropdown to choose pathway of lowest level\n",
    "                                            \"Available sub-pathway:    \",\n",
    "                                            color=\"white\",\n",
    "                                            text_color=\"muted\",\n",
    "                                            class_name=\"border me-1\",\n",
    "                                        ),\n",
    "                                        html.Div(style={'padding': 7}), #Space\n",
    "                                        dcc.Dropdown(\n",
    "                                            id=\"Lowest_Level\",\n",
    "                                            options=[{\"label\": name , \"value\": name }\n",
    "                                                     for name in path_list   #change\n",
    "                                                    ],\n",
    "                                            clearable=False,\n",
    "                                            value = path_list[0],             # 'AHR'\n",
    "                                            style={\"width\": \"187px\"},\n",
    "                                        )])]),\n",
    "\n",
    "                        html.H4(dbc.Badge(\n",
    "                            \"Model organism(s):\", color=\"success\", className=\"mr-1\"\n",
    "                        )),\n",
    "                        dcc.Dropdown(\n",
    "                            id=\"organism_dropdown\",\n",
    "                            options=[{\"label\": i + \" (\" + str(v)+ \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype.items()],\n",
    "                            value= [{\"label\": i + \" (\" + str(v)+ \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype.items()] ,    #check??\n",
    "                            #value= org_totalPhenotype.keys(),   #if change will cause layout error\n",
    "                            multi=True,\n",
    "                            style={\"width\": \"100%\"},\n",
    "                        ),\n",
    "                        html.Div(style={'padding': 7}),\n",
    "                        dbc.Button(\"Search\", id=\"search-button\", className=\"me-2\", n_clicks=0, size=\"sm\",color=\"secondary\"),\n",
    "                        html.Span(id=\"search-status\", style={\"verticalAlign\": \"middle\"}),\n",
    "                        html.Div(style={'padding': 20}),  #Space\n",
    "\n",
    "                        html.H4(dbc.Badge(\n",
    "                            \"Choose level of visualisation:\", color=\"success\", className=\"mr-1\"\n",
    "                        )),\n",
    "                        html.Div(style={'padding': 10}), #Space\n",
    "                        dcc.Slider(\n",
    "                            id='detail_slider',\n",
    "                            min=1,\n",
    "                            max=3,\n",
    "                            value=3,\n",
    "                            marks={1: \"human_gene\",2:\"organism_ortholog\", 3:\"organism_phenotype\"},\n",
    "                            step=1\n",
    "                        )\n",
    "                    ]\n",
    "                )\n",
    "                ]\n",
    "        )\n",
    "\n",
    "    ]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "1fec23bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_items([('dmelanogaster', 217), ('mouse', 350), ('celegans', 57), ('zebrafish', 0)])"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "org_totalPhenotype.items()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f98d2073",
   "metadata": {},
   "outputs": [],
   "source": [
    "app.layout = html.Div([navbar, body_layout])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecec7ec1",
   "metadata": {},
   "source": [
    "## Dash Callbacks"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c19917ae",
   "metadata": {},
   "source": [
    "### Click on a node to see its details here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "672dcd8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# call back Node information -----------------\n",
    "@app.callback(\n",
    "    Output('phenotype-data', 'children'),\n",
    "    [Input('cytoscape-phenotype', 'tapNodeData')]\n",
    ")\n",
    "def display_nodeData(data):\n",
    "    contents = 'Click on a node to see its details here'\n",
    "    if data:\n",
    "        contents = []\n",
    "        if data.get('indent') == 'phenotype':\n",
    "            contents.append(html.H5('Phenotype: ' + data.get('label')))\n",
    "            contents.append(html.P('Hierchically related phenotype(s): ' + str(data.get('metadata')[1])))\n",
    "            contents.append(html.P('Phenotype enrichment Q-value: ' + str(data.get('metadata')[2])))\n",
    "            contents.append(html.P('Phenotype enrichment P-value: ' + str(data.get('metadata')[4])))\n",
    "            return contents\n",
    "        elif data.get('indent') == 'gene':\n",
    "            contents.append(html.H5('Gene ID: ' + data.get('label')))\n",
    "            return contents\n",
    "        elif data.get('indent') == 'ortholog':\n",
    "            contents.append(html.H5('Ortholog ID: ' + data.get('label')))\n",
    "            contents.append(html.P('Organism: ' + data.get('organism')))\n",
    "            return contents\n",
    "    else:\n",
    "        return contents"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e799148",
   "metadata": {},
   "source": [
    "### Search phenotype for a gene:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a6aa3249",
   "metadata": {},
   "outputs": [],
   "source": [
    "#@app.callback(\n",
    "#     Output('phenotype-data', 'children'),\n",
    "#     [Input('cytoscape-phenotype', 'tapNodeData')]\n",
    "# )\n",
    "# def display_phenotype(data):\n",
    "#     contents = 'Click on a node to see its details here'\n",
    "#     if data:\n",
    "#         contents = []\n",
    "#         #contents.append(html.H5('Phenotype: ' + str(data.get('metadata')[0])))\n",
    "#         contents.append(html.H5('Phenotype: ' + data.get('id')))\n",
    "#         contents.append(html.P('Hierchically related phenotype(s): ' + str(data.get('metadata')[1])))\n",
    "#         contents.append(html.P('Phenotype enrichment Q-value: ' + str(data.get('metadata')[3])))\n",
    "#         contents.append(html.P('Phenotype enrichment P-value: ' + str(data.get('metadata')[4])))\n",
    "#         return contents\n",
    "#     else:\n",
    "#         return contents"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32945720",
   "metadata": {},
   "source": [
    "### Level of pathway:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "b191ba08",
   "metadata": {},
   "outputs": [],
   "source": [
    "@app.callback(\n",
    "    [Output('cytoscape-phenotype', \"elements\"),\n",
    "     Output('Lowest_Level','options'),\n",
    "     Output('organism_dropdown', 'value'),\n",
    "     ],\n",
    "    [Input('pathway_level', 'value')]\n",
    ")\n",
    "def filter_nodes_1(value):\n",
    "    if value == 1:\n",
    "        options = [{\"label\": name, \"value\": name}\n",
    "                   for name in path_list]   # change\n",
    "        value = [{\"label\": i + \" (\" + str(v) + \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype_1.items()]\n",
    "        return new_element1, options, value\n",
    "    if value == 2:\n",
    "        options = [{\"label\": name , \"value\": name } for name in ['']] #check\n",
    "        value = [{\"label\": i + \" (\" + str(v) + \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype.items()]\n",
    "        return element, options, value\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e128e24f",
   "metadata": {},
   "source": [
    "### Level of lowest pathway:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "bc0639d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#@app.callback(\n",
    "#    [Output('cytoscape-phenotype', \"elements\"),\n",
    "#     Output('organism_dropdown', 'value'),\n",
    "#     ],\n",
    "#    [Input('Lowest_Level', 'value')]\n",
    "#)\n",
    "\n",
    "#def filter_nodes_2(value):\n",
    "#    if value == path_list[0]:\n",
    "#        value = [{\"label\": i + \" (\" + str(v) + \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype_1.items()]\n",
    "#        return new_element1, value\n",
    "#    if value == path_list[1]:\n",
    "#        value = [{\"label\": i + \" (\" + str(v) + \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype_2.items()]\n",
    "#        return new_element2, value\n",
    "#    if value == path_list[2]:\n",
    "#        value = [{\"label\": i + \" (\" + str(v) + \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype_3.items()]\n",
    "#        return new_element3, value\n",
    "#    if value == path_list[3]:\n",
    "#        value = [{\"label\": i + \" (\" + str(v) + \" phenotype(s))\", \"value\": i} for i, v in org_totalPhenotype_4.items()]\n",
    "#        return new_element4, value"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bee8db8",
   "metadata": {},
   "source": [
    "### Model organism(s):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "cf356169",
   "metadata": {},
   "outputs": [],
   "source": [
    "@app.callback (\n",
    "    Output('cytoscape-phenotype', \"elements\"),\n",
    "    [Input('organism_dropdown', 'value'), \n",
    "    Input('Lowest_Level', 'value')]\n",
    ")\n",
    "\n",
    "def select_organism(value1, value2):\n",
    "    new_elements = []\n",
    "    if value1 == organism[0]:\n",
    "        if value2 == path_list[0]:\n",
    "            for i in new_elements1.get('data').get('organism') == 'dmelanogaster':\n",
    "                new_elements.append(i) \n",
    "        if value2 == path_list[1]:\n",
    "            for i in new_elements1.get('data').get('organism') == 'mouse':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[2]:\n",
    "            for i in new_elements1.get('data').get('organism') == 'celegans':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[3]:\n",
    "            for i in new_elements1.get('data').get('organism') == 'zebrafish':\n",
    "                new_elements.append(i)\n",
    "    if value1 == organism[1]:\n",
    "        if value2 == path_list[0]:\n",
    "            for i in new_elements2.get('data').get('organism') == 'dmelanogaster':\n",
    "                new_elements.append(i) \n",
    "        if value2 == path_list[1]:\n",
    "            for i in new_elements2.get('data').get('organism') == 'mouse':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[2]:\n",
    "            for i in new_elements2.get('data').get('organism') == 'celegans':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[3]:\n",
    "            for i in new_elements2.get('data').get('organism') == 'zebrafish':\n",
    "                new_elements.append(i)\n",
    "    if value1 == organism[2]:\n",
    "        if value2 == path_list[0]:\n",
    "            for i in new_elements3.get('data').get('organism') == 'dmelanogaster':\n",
    "                new_elements.append(i) \n",
    "        if value2 == path_list[1]:\n",
    "            for i in new_elements3.get('data').get('organism') == 'mouse':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[2]:\n",
    "            for i in new_elements3.get('data').get('organism') == 'celegans':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[3]:\n",
    "            for i in new_elements3.get('data').get('organism') == 'zebrafish':\n",
    "                new_elements.append(i)\n",
    "    if value1 == organism[3]:\n",
    "        if value2 == path_list[0]:\n",
    "            for i in new_elements4.get('data').get('organism') == 'dmelanogaster':\n",
    "                new_elements.append(i) \n",
    "        if value2 == path_list[1]:\n",
    "            for i in new_elements4.get('data').get('organism') == 'mouse':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[2]:\n",
    "            for i in new_elements4.get('data').get('organism') == 'celegans':\n",
    "                new_elements.append(i)\n",
    "        if value2 == path_list[3]:\n",
    "            for i in new_elements4.get('data').get('organism') == 'zebrafish':\n",
    "                new_elements.append(i)\n",
    "    return new_elements\n",
    "#-----------------------------------------------\n",
    "\n",
    "# call back Slider + Dropdown -----------------\n",
    "# @app.callback(\n",
    "#     Output('cytoscape-phenotype', \"elements\"),\n",
    "#     [\n",
    "#         Input('pathway_level', 'value'),\n",
    "#         Input('organism_dropdown', 'value'),\n",
    "#         Input('detail_slider', 'value'),\n",
    "#     ]\n",
    "# )\n",
    "# def filter_nodes(pathway_level, organism_list, detail_slider):\n",
    "#     ctx = dash.callback_context\n",
    "#     if not ctx.triggered:\n",
    "#         return element\n",
    "#\n",
    "#     else:\n",
    "#         trigger_id = ctx.triggered[0]['prop_id'].split(\".\")[0]\n",
    "#         val = ctx.triggered[0]['value'].split(\".\")[0]\n",
    "#\n",
    "#         if trigger_id == 'pathway_level':\n",
    "#             pathway = val\n",
    "#             df = gene_df1 if val == 1 else gene_df2\n",
    "#         elif trigger_id == 'organism_dropdown':\n",
    "\n",
    "\n",
    "#----------------------------------------------\n",
    "# @app.callback(\n",
    "#     Output('cytoscape-phenotype', 'elements'),\n",
    "#     [Input('pathway_level', 'value'),\n",
    "#      Input('organism_dropdown', 'value')]\n",
    "# )\n",
    "# def filter_nodes(pathway_level):\n",
    "#     if pathway_level == 1:\n",
    "#         metadata1 = db_retrieve.select_from_metadata(\"AHR\")\n",
    "#         new_nodes, new_edges = load_info_to_graph(gene_df_1, metadata1, N_genes_1)\n",
    "#         new_element = new_edges + new_nodes\n",
    "#         return new_element\n",
    "#\n",
    "#     return element\n",
    "#---------------------------------------------\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00dc9f04",
   "metadata": {},
   "source": [
    "### Running the app:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "91c51713",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      " * Serving Flask app '__main__' (lazy loading)\n",
      " * Environment: production\n",
      "\u001b[31m   WARNING: This is a development server. Do not use it in a production deployment.\u001b[0m\n",
      "\u001b[2m   Use a production WSGI server instead.\u001b[0m\n",
      " * Debug mode: off\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " * Running on http://127.0.0.1:8050/ (Press CTRL+C to quit)\n"
     ]
    }
   ],
   "source": [
    "if __name__ == '__main__':\n",
    "    app.run_server(debug=False)  # set false so you can load bigger networks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e8ff6f0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dd28da5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
